// @uxuissk/design-system/dist/sellsuki-ds.js
import { jsx as e, jsxs as s, Fragment as Qe } from "react/jsx-runtime";
import et, { forwardRef as ze, useState as z, useCallback as P, useRef as X, useContext as $t, createContext as zt, useEffect as V, useMemo as te } from "react";
import { Loader2 as Se, X as J, EyeOff as rr, Eye as nr, AlertCircle as ke, CheckCircle2 as se, XCircle as kt, AlertTriangle as Ft, Info as Wt, Minus as Ee, Check as fe, Calendar as or, ChevronsLeft as tt, ChevronLeft as $e, ChevronRight as le, ChevronsRight as rt, Clock as ar, MoreHorizontal as Lt, Search as Be, ChevronUp as sr, ChevronDown as Ye, ChevronsUpDown as lr, Slash as ir, Dot as cr, Copy as it, Inbox as dr, Plus as jt, Upload as Xe, Image as ur, Film as fr, FileText as mr, File as Ct, ZoomOut as hr, ZoomIn as br, RotateCw as gr, ExternalLink as pr, Bell as Mt, BellOff as vr, Heart as xr, ThumbsUp as yr, Star as wr, ArrowUpRight as Dt, ArrowDownRight as At, Circle as ct, Menu as Nr, FolderOpen as Sr, Folder as $r } from "lucide-react";
import { createPortal as nt } from "react-dom";
const zr = {
  sm: "h-8 px-3 gap-1.5",
  md: "h-10 px-4 gap-2",
  lg: "h-12 px-6 gap-2.5",
  xl: "h-14 px-8 gap-3"
}, kr = {
  sm: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)" },
  md: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)" },
  lg: { fontFamily: "var(--font-button)", fontSize: "var(--text-label)", fontWeight: "var(--weight-button)" },
  xl: { fontFamily: "var(--font-button)", fontSize: "var(--text-caption)", fontWeight: "var(--weight-button)" }
}, It = {
  sm: 14,
  md: 16,
  lg: 18,
  xl: 20
}, Tt = {
  primary: "bg-primary text-primary-foreground border border-primary hover:bg-primary/90 active:bg-primary/80 shadow-elevation-sm",
  secondary: "bg-secondary text-secondary-foreground border border-secondary hover:bg-secondary/90 active:bg-secondary/80 shadow-elevation-sm",
  outline: "bg-transparent text-foreground border border-border hover:bg-muted active:bg-muted/80",
  ghost: "bg-transparent text-foreground border border-transparent hover:bg-muted active:bg-muted/80",
  destructive: "bg-destructive text-destructive-foreground border border-destructive hover:bg-destructive/90 active:bg-destructive/80 shadow-elevation-sm",
  link: "bg-transparent text-primary border border-transparent underline-offset-4 hover:underline px-0 h-auto"
}, Fr = ze(
  ({
    variant: t = "primary",
    size: r = "md",
    loading: o = !1,
    loadingText: n,
    leftIcon: a,
    rightIcon: i,
    fullWidth: l = !1,
    active: c = !1,
    disabled: d,
    children: f,
    className: m = "",
    ...v
  }, p) => {
    const u = d || o;
    return /* @__PURE__ */ e(
      "button",
      {
        ref: p,
        disabled: u,
        className: `inline-flex items-center justify-center rounded-[var(--radius-md)] transition-all duration-150
          ${zr[r]}
          ${Tt[t]}
          ${l ? "w-full" : ""}
          ${c ? "ring-2 ring-ring/30" : ""}
          ${u ? "opacity-50 cursor-not-allowed pointer-events-none" : "cursor-pointer active:scale-[0.98]"}
          focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring/40
          ${m}
        `,
        style: kr[r],
        ...v,
        children: o ? /* @__PURE__ */ s(Qe, { children: [
          /* @__PURE__ */ e(Se, { size: It[r], className: "animate-spin flex-shrink-0" }),
          n && /* @__PURE__ */ e("span", { children: n })
        ] }) : /* @__PURE__ */ s(Qe, { children: [
          a && /* @__PURE__ */ e("span", { className: "flex-shrink-0", children: a }),
          f && /* @__PURE__ */ e("span", { children: f }),
          i && /* @__PURE__ */ e("span", { className: "flex-shrink-0", children: i })
        ] })
      }
    );
  }
);
Fr.displayName = "DSButton";
const Wr = ze(
  ({
    variant: t = "ghost",
    size: r = "md",
    loading: o = !1,
    icon: n,
    disabled: a,
    className: i = "",
    ...l
  }, c) => {
    const d = a || o;
    return /* @__PURE__ */ e(
      "button",
      {
        ref: c,
        disabled: d,
        className: `inline-flex items-center justify-center rounded-[var(--radius-md)] transition-all duration-150
          ${{
          sm: "w-8 h-8",
          md: "w-10 h-10",
          lg: "w-12 h-12",
          xl: "w-14 h-14"
        }[r]}
          ${Tt[t]}
          ${d ? "opacity-50 cursor-not-allowed pointer-events-none" : "cursor-pointer active:scale-[0.98]"}
          focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring/40
          ${i}
        `,
        ...l,
        children: o ? /* @__PURE__ */ e(Se, { size: It[r], className: "animate-spin" }) : n
      }
    );
  }
);
Wr.displayName = "IconButton";
function Bo({
  children: t,
  className: r = ""
}) {
  return /* @__PURE__ */ e(
    "div",
    {
      className: `inline-flex [&>button]:rounded-none [&>button:first-child]:rounded-l-[var(--radius-md)] [&>button:last-child]:rounded-r-[var(--radius-md)] [&>button:not(:last-child)]:border-r-0 ${r}`,
      children: t
    }
  );
}
const Et = {
  sm: {
    wrapper: "h-9 px-2.5 gap-1.5",
    textStyle: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
    icon: 14
  },
  md: {
    wrapper: "h-11 px-3 gap-2",
    textStyle: { fontFamily: "var(--font-caption)", fontSize: "var(--text-caption)", fontWeight: "var(--weight-caption)" },
    icon: 16
  },
  lg: {
    wrapper: "h-13 px-4 gap-2.5",
    textStyle: { fontFamily: "var(--font-p)", fontSize: "var(--text-p)", fontWeight: "var(--weight-p)" },
    icon: 18
  }
}, Bt = {
  default: {
    base: "border border-border bg-input-background",
    hover: "hover:border-foreground/30",
    focus: "ring-2 ring-ring/20 border-ring"
  },
  outlined: {
    base: "border-2 border-border bg-transparent",
    hover: "hover:border-foreground/40",
    focus: "ring-2 ring-ring/20 border-ring"
  },
  filled: {
    base: "border border-transparent bg-muted",
    hover: "hover:bg-muted/80",
    focus: "ring-2 ring-ring/20 border-ring bg-input-background"
  },
  ghost: {
    base: "border border-transparent bg-transparent",
    hover: "hover:bg-muted/50",
    focus: "ring-2 ring-ring/20 bg-input-background"
  }
}, Yt = {
  default: "",
  error: "!border-destructive ring-destructive/20",
  success: "!border-chart-2 ring-chart-2/20",
  warning: "!border-chart-5 ring-chart-5/20"
}, Ie = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, Lr = ze(
  ({
    label: t,
    helperText: r,
    errorMessage: o,
    successMessage: n,
    inputSize: a = "md",
    variant: i = "default",
    state: l = "default",
    leftIcon: c,
    rightIcon: d,
    prefix: f,
    suffix: m,
    clearable: v = !1,
    showPasswordToggle: p = !1,
    loading: u = !1,
    fullWidth: S = !0,
    required: N = !1,
    onClear: h,
    disabled: x,
    type: D = "text",
    value: b,
    defaultValue: y,
    onChange: $,
    className: k = "",
    ...M
  }, I) => {
    const [W, L] = z(!1), [Y, F] = z(!1), [w, T] = z(y ?? ""), H = b !== void 0 ? b : w, U = Et[a], Z = Bt[i], K = o ? "error" : l, Q = D === "password", O = Q && Y ? "text" : D, ne = P(
      (me) => {
        T(me.target.value), $ == null || $(me);
      },
      [$]
    ), oe = P(() => {
      T(""), h == null || h();
    }, [h]), ae = String(H).length > 0;
    return /* @__PURE__ */ s("div", { className: `${S ? "w-full" : "inline-flex flex-col"} ${k}`, children: [
      t && /* @__PURE__ */ s("label", { className: "block mb-1.5 text-foreground", children: [
        t,
        N && /* @__PURE__ */ e("span", { className: "text-destructive ml-0.5", children: "*" })
      ] }),
      /* @__PURE__ */ s(
        "div",
        {
          className: `flex items-center rounded-[var(--radius-md)] transition-all duration-150
            ${U.wrapper}
            ${Z.base}
            ${x ? "" : Z.hover}
            ${W && !x ? Z.focus : ""}
            ${x ? "opacity-50 cursor-not-allowed" : ""}
            ${Yt[K]}
          `,
          children: [
            c && /* @__PURE__ */ e("span", { className: "flex-shrink-0 text-muted-foreground", children: c }),
            f && /* @__PURE__ */ e("span", { className: "flex-shrink-0 text-muted-foreground", style: Ie, children: f }),
            /* @__PURE__ */ e(
              "input",
              {
                ref: I,
                type: O,
                disabled: x || u,
                value: H,
                onChange: ne,
                onFocus: () => L(!0),
                onBlur: () => L(!1),
                className: "flex-1 min-w-0 bg-transparent border-none outline-none text-foreground placeholder:text-muted-foreground disabled:cursor-not-allowed",
                style: U.textStyle,
                ...M
              }
            ),
            m && /* @__PURE__ */ e("span", { className: "flex-shrink-0 text-muted-foreground", style: Ie, children: m }),
            v && ae && !x && /* @__PURE__ */ e(
              "button",
              {
                type: "button",
                onClick: oe,
                className: "flex-shrink-0 text-muted-foreground hover:text-foreground transition-colors cursor-pointer p-0.5",
                children: /* @__PURE__ */ e(J, { size: U.icon - 2 })
              }
            ),
            Q && p && /* @__PURE__ */ e(
              "button",
              {
                type: "button",
                onClick: () => F(!Y),
                className: "flex-shrink-0 text-muted-foreground hover:text-foreground transition-colors cursor-pointer p-0.5",
                children: Y ? /* @__PURE__ */ e(rr, { size: U.icon }) : /* @__PURE__ */ e(nr, { size: U.icon })
              }
            ),
            d && !Q && /* @__PURE__ */ e("span", { className: "flex-shrink-0 text-muted-foreground", children: d }),
            u && /* @__PURE__ */ e("span", { className: "flex-shrink-0", children: /* @__PURE__ */ e("span", { className: "block w-4 h-4 border-2 border-muted-foreground border-t-transparent rounded-full animate-spin" }) })
          ]
        }
      ),
      (r || o || n) && /* @__PURE__ */ s("div", { className: "mt-1.5 flex items-center gap-1", children: [
        K === "error" && /* @__PURE__ */ e(ke, { size: 12, className: "text-destructive flex-shrink-0" }),
        K === "success" && /* @__PURE__ */ e(se, { size: 12, className: "text-chart-2 flex-shrink-0" }),
        /* @__PURE__ */ e(
          "span",
          {
            className: `${K === "error" ? "text-destructive" : K === "success" ? "text-chart-2" : "text-muted-foreground"}`,
            style: Ie,
            children: o || n || r
          }
        )
      ] })
    ] });
  }
);
Lr.displayName = "DSInput";
const jr = ze(
  ({
    label: t,
    helperText: r,
    errorMessage: o,
    successMessage: n,
    inputSize: a = "md",
    variant: i = "default",
    state: l = "default",
    showCharCount: c = !1,
    maxLength: d,
    required: f = !1,
    disabled: m,
    value: v,
    defaultValue: p,
    onChange: u,
    className: S = "",
    ...N
  }, h) => {
    const [x, D] = z(!1), [b, y] = z(String(p ?? "").length), $ = Et[a], k = Bt[i], M = o ? "error" : l, I = P(
      (W) => {
        y(W.target.value.length), u == null || u(W);
      },
      [u]
    );
    return /* @__PURE__ */ s("div", { className: `w-full ${S}`, children: [
      t && /* @__PURE__ */ s("label", { className: "block mb-1.5 text-foreground", children: [
        t,
        f && /* @__PURE__ */ e("span", { className: "text-destructive ml-0.5", children: "*" })
      ] }),
      /* @__PURE__ */ s(
        "div",
        {
          className: `transition-all duration-150 overflow-hidden rounded-[var(--radius-md)]
            ${k.base}
            ${m ? "" : k.hover}
            ${x && !m ? k.focus : ""}
            ${m ? "opacity-50 cursor-not-allowed" : ""}
            ${Yt[M]}
          `,
          children: [
            /* @__PURE__ */ e(
              "textarea",
              {
                ref: h,
                disabled: m,
                value: v,
                defaultValue: p,
                maxLength: d,
                onChange: I,
                onFocus: () => D(!0),
                onBlur: () => D(!1),
                className: "w-full min-h-[100px] px-3 py-2.5 bg-transparent border-none outline-none text-foreground placeholder:text-muted-foreground resize-y disabled:cursor-not-allowed",
                style: $.textStyle,
                ...N
              }
            ),
            c && /* @__PURE__ */ s(
              "div",
              {
                className: "flex justify-end px-3 py-1.5 border-t border-border text-muted-foreground bg-muted/20",
                style: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" },
                children: [
                  b,
                  d ? ` / ${d}` : "",
                  " chars"
                ]
              }
            )
          ]
        }
      ),
      (r || o || n) && /* @__PURE__ */ s("div", { className: "mt-1.5 flex items-center gap-1", children: [
        M === "error" && /* @__PURE__ */ e(ke, { size: 12, className: "text-destructive flex-shrink-0" }),
        M === "success" && /* @__PURE__ */ e(se, { size: 12, className: "text-chart-2 flex-shrink-0" }),
        /* @__PURE__ */ e(
          "span",
          {
            className: `${M === "error" ? "text-destructive" : M === "success" ? "text-chart-2" : "text-muted-foreground"}`,
            style: Ie,
            children: o || n || r
          }
        )
      ] })
    ] });
  }
);
jr.displayName = "DSTextarea";
const Cr = {
  info: { bg: "bg-primary/5", border: "border-primary/30", text: "text-primary", icon: /* @__PURE__ */ e(Wt, { size: 18 }) },
  success: { bg: "bg-chart-2/5", border: "border-chart-2/30", text: "text-chart-2", icon: /* @__PURE__ */ e(se, { size: 18 }) },
  warning: { bg: "bg-chart-5/5", border: "border-chart-5/30", text: "text-chart-5", icon: /* @__PURE__ */ e(Ft, { size: 18 }) },
  error: { bg: "bg-destructive/5", border: "border-destructive/30", text: "text-destructive", icon: /* @__PURE__ */ e(kt, { size: 18 }) }
};
function Mr({ variant: t = "info", title: r, children: o, dismissible: n = !1, onDismiss: a, action: i, icon: l, className: c = "" }) {
  const [d, f] = z(!0), m = Cr[t];
  if (!d) return null;
  const v = () => {
    f(!1), a == null || a();
  };
  return /* @__PURE__ */ s("div", { className: `flex gap-3 p-4 rounded-[var(--radius)] border ${m.bg} ${m.border} ${c}`, role: "alert", children: [
    /* @__PURE__ */ e("span", { className: `flex-shrink-0 mt-0.5 ${m.text}`, children: l ?? m.icon }),
    /* @__PURE__ */ s("div", { className: "flex-1 min-w-0", children: [
      r && /* @__PURE__ */ e("span", { className: `block ${m.text}`, style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-button)" }, children: r }),
      /* @__PURE__ */ e("span", { className: "text-foreground block", style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" }, children: o }),
      i && /* @__PURE__ */ e("div", { className: "mt-2", children: i })
    ] }),
    n && /* @__PURE__ */ e("button", { onClick: v, className: "flex-shrink-0 text-muted-foreground hover:text-foreground transition-colors cursor-pointer", "aria-label": "Dismiss", children: /* @__PURE__ */ e(J, { size: 16 }) })
  ] });
}
let _t = [], we = [];
function dt() {
  _t.forEach((t) => t([...we]));
}
const ve = {
  show: (t) => {
    const r = { ...t, id: `toast-${Date.now()}-${Math.random().toString(36).slice(2)}` };
    we = [...we, r], dt(), setTimeout(() => {
      ve.dismiss(r.id);
    }, t.duration ?? 4e3);
  },
  info: (t, r) => ve.show({ variant: "info", message: t, title: r }),
  success: (t, r) => ve.show({ variant: "success", message: t, title: r }),
  warning: (t, r) => ve.show({ variant: "warning", message: t, title: r }),
  error: (t, r) => ve.show({ variant: "error", message: t, title: r }),
  dismiss: (t) => {
    we = we.filter((r) => r.id !== t), dt();
  }
};
function Yo() {
  const [t, r] = z([]), o = X(!1);
  return o.current || (o.current = !0, _t.push(r)), t.length === 0 ? null : /* @__PURE__ */ e("div", { className: "fixed bottom-4 right-4 z-[200] flex flex-col gap-2 max-w-sm w-full pointer-events-none", children: t.map((n) => /* @__PURE__ */ e("div", { className: "pointer-events-auto animate-in slide-in-from-right fade-in duration-300", children: /* @__PURE__ */ e(Mr, { variant: n.variant, title: n.title, dismissible: !0, onDismiss: () => ve.dismiss(n.id), children: n.message }) }, n.id)) });
}
const Dr = {
  default: "bg-primary text-primary-foreground",
  secondary: "bg-secondary/15 text-secondary",
  outline: "bg-transparent border border-border text-foreground",
  destructive: "bg-destructive/15 text-destructive",
  success: "bg-chart-2/15 text-chart-2",
  warning: "bg-chart-5/15 text-chart-5"
}, Ar = {
  sm: {
    className: "px-1.5 py-0.5 gap-1",
    style: {
      fontFamily: "var(--font-button)",
      fontSize: "var(--text-button)",
      fontWeight: "var(--weight-button)"
    },
    iconSize: 10
  },
  md: {
    className: "px-2.5 py-0.5 gap-1.5",
    style: {
      fontFamily: "var(--font-button)",
      fontSize: "var(--text-button)",
      fontWeight: "var(--weight-button)"
    },
    iconSize: 12
  },
  lg: {
    className: "px-3 py-1 gap-1.5",
    style: {
      fontFamily: "var(--font-button)",
      fontSize: "var(--text-button)",
      fontWeight: "var(--weight-button)"
    },
    iconSize: 14
  }
}, Ir = {
  default: "bg-primary-foreground",
  secondary: "bg-secondary",
  outline: "bg-foreground",
  destructive: "bg-destructive",
  success: "bg-chart-2",
  warning: "bg-chart-5"
};
function _o({
  children: t,
  variant: r = "default",
  size: o = "md",
  icon: n,
  dot: a,
  removable: i,
  onRemove: l,
  className: c = ""
}) {
  const d = Ar[o];
  return /* @__PURE__ */ s(
    "span",
    {
      className: `inline-flex items-center rounded-full whitespace-nowrap ${Dr[r]} ${d.className} ${c}`,
      style: { ...d.style, lineHeight: "1.2" },
      children: [
        a && /* @__PURE__ */ e(
          "span",
          {
            className: `w-1.5 h-1.5 rounded-full flex-shrink-0 ${Ir[r]}`
          }
        ),
        n && /* @__PURE__ */ e("span", { className: "flex-shrink-0", children: n }),
        /* @__PURE__ */ e("span", { children: t }),
        i && /* @__PURE__ */ e(
          "button",
          {
            onClick: l,
            className: "flex-shrink-0 hover:opacity-70 transition-opacity cursor-pointer rounded-full -mr-0.5",
            "aria-label": "Remove",
            children: /* @__PURE__ */ e(J, { size: d.iconSize })
          }
        )
      ]
    }
  );
}
const Tr = {
  sm: {
    box: "w-4 h-4",
    icon: 10,
    labelStyle: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" },
    descStyle: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" }
  },
  md: {
    box: "w-5 h-5",
    icon: 12,
    labelStyle: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
    descStyle: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" }
  },
  lg: {
    box: "w-6 h-6",
    icon: 14,
    labelStyle: { fontFamily: "var(--font-caption)", fontSize: "var(--text-caption)", fontWeight: "var(--weight-label)" },
    descStyle: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" }
  }
};
function Ro({
  checked: t,
  indeterminate: r = !1,
  defaultChecked: o = !1,
  onChange: n,
  label: a,
  description: i,
  size: l = "md",
  disabled: c = !1,
  error: d,
  className: f = "",
  id: m
}) {
  const [v, p] = z(o), u = t ?? v, S = Tr[l], N = m ?? `cb-${et.useId()}`, h = () => {
    if (c) return;
    const D = !u;
    t === void 0 && p(D), n == null || n(D);
  }, x = u || r;
  return /* @__PURE__ */ s("div", { className: `inline-flex items-start gap-2.5 ${c ? "opacity-50" : ""} ${f}`, children: [
    /* @__PURE__ */ e(
      "button",
      {
        id: N,
        role: "checkbox",
        "aria-checked": r ? "mixed" : u,
        disabled: c,
        onClick: h,
        className: `${S.box} flex-shrink-0 rounded-[var(--radius-sm)] border-2 transition-all flex items-center justify-center cursor-pointer mt-0.5
          ${x ? "bg-primary border-primary text-primary-foreground" : d ? "border-destructive bg-card" : "border-border bg-card hover:border-primary/50"}
          ${c ? "!cursor-not-allowed" : ""}
          focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 outline-none`,
        children: r ? /* @__PURE__ */ e(Ee, { size: S.icon, strokeWidth: 3 }) : u ? /* @__PURE__ */ e(fe, { size: S.icon, strokeWidth: 3 }) : null
      }
    ),
    (a || i) && /* @__PURE__ */ s("div", { className: "min-w-0", children: [
      a && /* @__PURE__ */ e(
        "label",
        {
          htmlFor: N,
          className: `text-foreground block cursor-pointer ${c ? "!cursor-not-allowed" : ""}`,
          style: S.labelStyle,
          onClick: h,
          children: a
        }
      ),
      i && /* @__PURE__ */ e("span", { className: "text-muted-foreground block", style: S.descStyle, children: i }),
      d && /* @__PURE__ */ e("span", { className: "text-destructive block mt-0.5", style: S.descStyle, children: d })
    ] })
  ] });
}
function Ho({ label: t, children: r, direction: o = "vertical", className: n = "" }) {
  return /* @__PURE__ */ s("fieldset", { className: `${n}`, children: [
    t && /* @__PURE__ */ e("legend", { className: "text-foreground mb-2 block", style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-button)" }, children: t }),
    /* @__PURE__ */ e("div", { className: `flex ${o === "vertical" ? "flex-col gap-3" : "flex-wrap gap-4"}`, children: r })
  ] });
}
const Rt = zt(null), Er = {
  sm: {
    outer: "w-4 h-4",
    inner: "w-1.5 h-1.5",
    labelStyle: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" },
    descStyle: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" }
  },
  md: {
    outer: "w-5 h-5",
    inner: "w-2 h-2",
    labelStyle: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
    descStyle: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" }
  },
  lg: {
    outer: "w-6 h-6",
    inner: "w-2.5 h-2.5",
    labelStyle: { fontFamily: "var(--font-caption)", fontSize: "var(--text-caption)", fontWeight: "var(--weight-label)" },
    descStyle: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" }
  }
};
function Oo({ value: t, label: r, description: o, disabled: n, size: a, className: i = "" }) {
  const l = $t(Rt), c = l ? l.value === t : !1, d = n ?? (l == null ? void 0 : l.disabled) ?? !1, f = a ?? (l == null ? void 0 : l.size) ?? "md", m = Er[f], v = () => {
    d || l == null || l.onChange(t);
  };
  return /* @__PURE__ */ s("div", { className: `inline-flex items-start gap-2.5 ${d ? "opacity-50" : ""} ${i}`, children: [
    /* @__PURE__ */ e(
      "button",
      {
        role: "radio",
        "aria-checked": c,
        disabled: d,
        onClick: v,
        className: `${m.outer} flex-shrink-0 rounded-full border-2 transition-all flex items-center justify-center cursor-pointer mt-0.5
          ${c ? "border-primary" : "border-border hover:border-primary/50"}
          ${d ? "!cursor-not-allowed" : ""}
          focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 outline-none`,
        children: c && /* @__PURE__ */ e("span", { className: `${m.inner} rounded-full bg-primary transition-transform scale-100` })
      }
    ),
    (r || o) && /* @__PURE__ */ s("div", { className: "min-w-0", children: [
      r && /* @__PURE__ */ e(
        "span",
        {
          className: `text-foreground block cursor-pointer ${d ? "!cursor-not-allowed" : ""}`,
          style: m.labelStyle,
          onClick: v,
          children: r
        }
      ),
      o && /* @__PURE__ */ e("span", { className: "text-muted-foreground block", style: m.descStyle, children: o })
    ] })
  ] });
}
function Uo({
  name: t,
  value: r,
  defaultValue: o = "",
  onChange: n,
  label: a,
  size: i = "md",
  disabled: l = !1,
  direction: c = "vertical",
  children: d,
  error: f,
  className: m = ""
}) {
  const [v, p] = z(o), u = r ?? v, S = (N) => {
    r === void 0 && p(N), n == null || n(N);
  };
  return /* @__PURE__ */ e(Rt.Provider, { value: { value: u, onChange: S, size: i, disabled: l, name: t }, children: /* @__PURE__ */ s("fieldset", { className: m, role: "radiogroup", children: [
    a && /* @__PURE__ */ e("legend", { className: "text-foreground mb-2 block", style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-button)" }, children: a }),
    /* @__PURE__ */ e("div", { className: `flex ${c === "vertical" ? "flex-col gap-3" : "flex-wrap gap-4"}`, children: d }),
    f && /* @__PURE__ */ e("span", { className: "text-destructive block mt-1.5", style: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" }, children: f })
  ] }) });
}
const Br = ["Su", "Mo", "Tu", "We", "Th", "Fr", "Sa"], Ge = [
  "January",
  "February",
  "March",
  "April",
  "May",
  "June",
  "July",
  "August",
  "September",
  "October",
  "November",
  "December"
];
function Ne(t, r) {
  return t.getFullYear() === r.getFullYear() && t.getMonth() === r.getMonth() && t.getDate() === r.getDate();
}
function ut(t, r, o) {
  if (!r || !o) return !1;
  const n = t.getTime();
  return n > r.getTime() && n < o.getTime();
}
function ft(t, r, o, n) {
  return !!(r && t < new Date(r.getFullYear(), r.getMonth(), r.getDate()) || o && t > new Date(o.getFullYear(), o.getMonth(), o.getDate()) || n != null && n.some((a) => Ne(a, t)));
}
function Yr(t, r) {
  return new Date(t, r + 1, 0).getDate();
}
function mt(t, r) {
  const o = new Date(t, r, 1).getDay(), n = Yr(t, r), a = [];
  for (let i = 0; i < o; i++) a.push(null);
  for (let i = 1; i <= n; i++) a.push(new Date(t, r, i));
  return a;
}
function Me(t, r) {
  if (!t) return "";
  const o = String(t.getDate()).padStart(2, "0"), n = String(t.getMonth() + 1).padStart(2, "0"), a = t.getFullYear();
  let i = `${o}/${n}/${a}`;
  if (r) {
    const l = String(t.getHours()).padStart(2, "0"), c = String(t.getMinutes()).padStart(2, "0");
    i += ` ${l}:${c}`;
  }
  return i;
}
const _r = {
  sm: {
    trigger: "h-8 px-2.5",
    text: {
      fontFamily: "var(--font-label)",
      fontSize: "var(--text-label)",
      fontWeight: "var(--weight-label)"
    },
    icon: 14,
    cell: "w-7 h-7"
  },
  md: {
    trigger: "h-10 px-3",
    text: {
      fontFamily: "var(--font-label)",
      fontSize: "var(--text-label)",
      fontWeight: "var(--weight-label)"
    },
    icon: 16,
    cell: "w-9 h-9"
  },
  lg: {
    trigger: "h-12 px-4",
    text: {
      fontFamily: "var(--font-caption)",
      fontSize: "var(--text-caption)",
      fontWeight: "var(--weight-caption)"
    },
    icon: 18,
    cell: "w-10 h-10"
  }
}, Rr = {
  default: "bg-input-background border border-border",
  outlined: "bg-transparent border-2 border-border",
  filled: "bg-muted/30 border border-transparent",
  ghost: "bg-transparent border border-transparent"
}, Hr = {
  default: "focus-within:ring-2 focus-within:ring-ring/30 focus-within:border-primary",
  error: "ring-2 ring-destructive/30 border-destructive",
  success: "ring-2 ring-chart-2/30 border-chart-2"
};
function Ko({
  value: t = null,
  rangeValue: r,
  onChange: o,
  onRangeChange: n,
  mode: a = "single",
  label: i,
  placeholder: l,
  helperText: c,
  errorMessage: d,
  successMessage: f,
  size: m = "md",
  variant: v = "default",
  state: p = "default",
  disabled: u = !1,
  required: S = !1,
  clearable: N = !1,
  showTime: h = !1,
  minDate: x,
  maxDate: D,
  disabledDates: b,
  showToday: y = !0,
  footer: $,
  fullWidth: k = !1
}) {
  const [M, I] = z(!1), [W, L] = z(() => t ?? /* @__PURE__ */ new Date()), [Y, F] = z("days"), [w, T] = z(null), [H, U] = z(() => t ? t.getHours() : 0), [Z, K] = z(() => t ? t.getMinutes() : 0), Q = X(null), O = _r[m];
  V(() => {
    function C(B) {
      Q.current && !Q.current.contains(B.target) && (I(!1), F("days"));
    }
    return M && document.addEventListener("mousedown", C), () => document.removeEventListener("mousedown", C);
  }, [M]), te(
    () => mt(W.getFullYear(), W.getMonth()),
    [W]
  );
  const ne = te(() => /* @__PURE__ */ new Date(), []), oe = () => L(new Date(W.getFullYear(), W.getMonth() - 1, 1)), ae = () => L(new Date(W.getFullYear(), W.getMonth() + 1, 1)), me = () => L(new Date(W.getFullYear() - 1, W.getMonth(), 1)), ce = () => L(new Date(W.getFullYear() + 1, W.getMonth(), 1)), E = P(
    (C) => {
      if (!ft(C, x, D, b))
        if (a === "single") {
          let B = new Date(C);
          h && B.setHours(H, Z), o == null || o(B), h || (I(!1), F("days"));
        } else {
          const B = r == null ? void 0 : r.start, ee = r == null ? void 0 : r.end;
          !B || B && ee ? n == null || n({ start: C, end: null }) : (C < B ? n == null || n({ start: C, end: B }) : n == null || n({ start: B, end: C }), I(!1), F("days"));
        }
    },
    [a, o, n, r, x, D, b, h, H, Z]
  ), ie = (C) => {
    C.stopPropagation(), a === "single" ? o == null || o(null) : n == null || n({ start: null, end: null });
  }, xe = () => {
    const C = /* @__PURE__ */ new Date();
    if (L(C), a === "single") {
      const B = new Date(C);
      h && B.setHours(H, Z), o == null || o(B), h || (I(!1), F("days"));
    }
  }, Fe = () => {
    if (t) {
      const C = new Date(t);
      C.setHours(H, Z), o == null || o(C);
    }
    I(!1), F("days");
  }, de = te(() => a === "single" ? Me(t, h) : r != null && r.start && (r != null && r.end) ? `${Me(r.start)} – ${Me(r.end)}` : r != null && r.start ? `${Me(r.start)} – ...` : "", [a, t, r, h]), ue = a === "single" ? !!t : !!(r != null && r.start || r != null && r.end), _e = p === "error" ? /* @__PURE__ */ e(ke, { size: O.icon, className: "text-destructive flex-shrink-0" }) : p === "success" ? /* @__PURE__ */ e(se, { size: O.icon, className: "text-chart-2 flex-shrink-0" }) : null, _ = p === "error" ? d : p === "success" ? f : c, ye = p === "error" ? "text-destructive" : p === "success" ? "text-chart-2" : "text-muted-foreground", Re = {
    fontFamily: "var(--font-label)",
    fontSize: "var(--text-label)",
    fontWeight: "var(--weight-label)"
  }, re = {
    fontFamily: "var(--font-button)",
    fontSize: "var(--text-button)",
    fontWeight: "var(--weight-button)"
  }, G = {
    fontFamily: "var(--font-button)",
    fontSize: "var(--text-button)",
    fontWeight: "var(--weight-button)"
  }, he = Math.floor(W.getFullYear() / 12) * 12;
  return /* @__PURE__ */ s("div", { className: `relative ${k ? "w-full" : ""}`, ref: Q, children: [
    i && /* @__PURE__ */ s("label", { className: "block text-foreground mb-1.5", style: Re, children: [
      i,
      S && /* @__PURE__ */ e("span", { className: "text-destructive ml-0.5", children: "*" })
    ] }),
    /* @__PURE__ */ s(
      "button",
      {
        type: "button",
        onClick: () => !u && I(!M),
        disabled: u,
        className: `
          w-full flex items-center gap-2 rounded-[var(--radius-md)] transition-all cursor-pointer
          ${O.trigger}
          ${Rr[v]}
          ${Hr[p]}
          ${u ? "opacity-50 cursor-not-allowed" : "hover:border-primary/50"}
        `,
        style: O.text,
        children: [
          /* @__PURE__ */ e(or, { size: O.icon, className: "text-muted-foreground flex-shrink-0" }),
          /* @__PURE__ */ e("span", { className: `flex-1 text-left truncate ${ue ? "text-foreground" : "text-muted-foreground"}`, children: ue ? de : l || (a === "range" ? "Select date range…" : "Select date…") }),
          _e,
          N && ue && !u && /* @__PURE__ */ e(
            "span",
            {
              onClick: ie,
              className: "text-muted-foreground hover:text-foreground cursor-pointer",
              children: /* @__PURE__ */ e(J, { size: O.icon - 2 })
            }
          )
        ]
      }
    ),
    _ && /* @__PURE__ */ e("p", { className: `mt-1.5 ${ye}`, style: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" }, children: _ }),
    M && /* @__PURE__ */ s(
      "div",
      {
        className: "absolute z-50 mt-1.5 bg-popover border border-border rounded-[var(--radius-lg)] shadow-elevation-sm overflow-hidden",
        style: { minWidth: a === "range" ? "560px" : "300px" },
        children: [
          /* @__PURE__ */ s("div", { className: "flex items-center justify-between px-4 py-3 border-b border-border bg-card", children: [
            /* @__PURE__ */ s("div", { className: "flex items-center gap-1", children: [
              /* @__PURE__ */ e("button", { onClick: me, className: "p-1 rounded-[var(--radius-sm)] hover:bg-muted text-muted-foreground hover:text-foreground transition-colors cursor-pointer", title: "Previous year", children: /* @__PURE__ */ e(tt, { size: 16 }) }),
              /* @__PURE__ */ e("button", { onClick: oe, className: "p-1 rounded-[var(--radius-sm)] hover:bg-muted text-muted-foreground hover:text-foreground transition-colors cursor-pointer", title: "Previous month", children: /* @__PURE__ */ e($e, { size: 16 }) })
            ] }),
            /* @__PURE__ */ s("div", { className: "flex items-center gap-2", children: [
              /* @__PURE__ */ e(
                "button",
                {
                  onClick: () => F(Y === "months" ? "days" : "months"),
                  className: "px-2 py-1 rounded-[var(--radius-sm)] hover:bg-muted text-foreground transition-colors cursor-pointer",
                  style: G,
                  children: Ge[W.getMonth()]
                }
              ),
              /* @__PURE__ */ e(
                "button",
                {
                  onClick: () => F(Y === "years" ? "days" : "years"),
                  className: "px-2 py-1 rounded-[var(--radius-sm)] hover:bg-muted text-foreground transition-colors cursor-pointer",
                  style: G,
                  children: W.getFullYear()
                }
              )
            ] }),
            /* @__PURE__ */ s("div", { className: "flex items-center gap-1", children: [
              /* @__PURE__ */ e("button", { onClick: ae, className: "p-1 rounded-[var(--radius-sm)] hover:bg-muted text-muted-foreground hover:text-foreground transition-colors cursor-pointer", title: "Next month", children: /* @__PURE__ */ e(le, { size: 16 }) }),
              /* @__PURE__ */ e("button", { onClick: ce, className: "p-1 rounded-[var(--radius-sm)] hover:bg-muted text-muted-foreground hover:text-foreground transition-colors cursor-pointer", title: "Next year", children: /* @__PURE__ */ e(rt, { size: 16 }) })
            ] })
          ] }),
          Y === "days" && /* @__PURE__ */ e("div", { className: `p-3 ${a === "range" ? "flex gap-3" : ""}`, children: (a === "range" ? [0, 1] : [0]).map((C) => {
            const B = new Date(W.getFullYear(), W.getMonth() + C, 1), ee = mt(B.getFullYear(), B.getMonth());
            return /* @__PURE__ */ s("div", { className: "flex-1", children: [
              C > 0 && /* @__PURE__ */ s("div", { className: "text-center mb-2 text-foreground", style: G, children: [
                Ge[B.getMonth()],
                " ",
                B.getFullYear()
              ] }),
              /* @__PURE__ */ e("div", { className: "grid grid-cols-7 mb-1", children: Br.map((R) => /* @__PURE__ */ e(
                "div",
                {
                  className: "text-center text-muted-foreground py-1",
                  style: re,
                  children: R
                },
                `${C}-${R}`
              )) }),
              /* @__PURE__ */ e("div", { className: "grid grid-cols-7", children: ee.map((R, We) => {
                if (!R) return /* @__PURE__ */ e("div", {}, `empty-${C}-${We}`);
                const Le = ft(R, x, D, b), He = Ne(R, ne), be = a === "single" ? t && Ne(R, t) : (r == null ? void 0 : r.start) && Ne(R, r.start) || (r == null ? void 0 : r.end) && Ne(R, r.end), Oe = a === "range" && (ut(R, (r == null ? void 0 : r.start) ?? null, (r == null ? void 0 : r.end) ?? null) || (r == null ? void 0 : r.start) && !(r != null && r.end) && w && ut(
                  R,
                  r.start,
                  w > r.start ? w : null
                ));
                return /* @__PURE__ */ e(
                  "button",
                  {
                    onClick: () => E(R),
                    onMouseEnter: () => a === "range" && T(R),
                    disabled: Le,
                    className: `
                              ${O.cell} flex items-center justify-center rounded-[var(--radius-sm)] transition-all cursor-pointer
                              ${Le ? "opacity-30 cursor-not-allowed" : "hover:bg-muted"}
                              ${be ? "bg-primary text-primary-foreground hover:bg-primary/90" : ""}
                              ${Oe && !be ? "bg-accent text-accent-foreground" : ""}
                              ${He && !be ? "border border-primary text-primary" : ""}
                            `,
                    style: G,
                    children: R.getDate()
                  },
                  `${C}-${R.getTime()}`
                );
              }) })
            ] }, C);
          }) }),
          Y === "months" && /* @__PURE__ */ e("div", { className: "p-3 grid grid-cols-3 gap-2", children: Ge.map((C, B) => /* @__PURE__ */ e(
            "button",
            {
              onClick: () => {
                L(new Date(W.getFullYear(), B, 1)), F("days");
              },
              className: `
                    py-2.5 rounded-[var(--radius-md)] transition-all cursor-pointer
                    ${W.getMonth() === B ? "bg-primary text-primary-foreground" : "text-foreground hover:bg-muted"}
                  `,
              style: G,
              children: C.substring(0, 3)
            },
            C
          )) }),
          Y === "years" && /* @__PURE__ */ s("div", { className: "p-3", children: [
            /* @__PURE__ */ s("div", { className: "flex items-center justify-between mb-3", children: [
              /* @__PURE__ */ e(
                "button",
                {
                  onClick: () => L(new Date(W.getFullYear() - 12, W.getMonth(), 1)),
                  className: "p-1 rounded-[var(--radius-sm)] hover:bg-muted text-muted-foreground hover:text-foreground cursor-pointer",
                  children: /* @__PURE__ */ e($e, { size: 16 })
                }
              ),
              /* @__PURE__ */ s("span", { className: "text-foreground", style: G, children: [
                he,
                " – ",
                he + 11
              ] }),
              /* @__PURE__ */ e(
                "button",
                {
                  onClick: () => L(new Date(W.getFullYear() + 12, W.getMonth(), 1)),
                  className: "p-1 rounded-[var(--radius-sm)] hover:bg-muted text-muted-foreground hover:text-foreground cursor-pointer",
                  children: /* @__PURE__ */ e(le, { size: 16 })
                }
              )
            ] }),
            /* @__PURE__ */ e("div", { className: "grid grid-cols-4 gap-2", children: Array.from({ length: 12 }, (C, B) => he + B).map((C) => /* @__PURE__ */ e(
              "button",
              {
                onClick: () => {
                  L(new Date(C, W.getMonth(), 1)), F("months");
                },
                className: `
                      py-2.5 rounded-[var(--radius-md)] transition-all cursor-pointer
                      ${W.getFullYear() === C ? "bg-primary text-primary-foreground" : "text-foreground hover:bg-muted"}
                      ${C === ne.getFullYear() && W.getFullYear() !== C ? "border border-primary text-primary" : ""}
                    `,
                style: G,
                children: C
              },
              C
            )) })
          ] }),
          h && a === "single" && Y === "days" && /* @__PURE__ */ s("div", { className: "px-4 py-3 border-t border-border flex items-center gap-3", children: [
            /* @__PURE__ */ e(ar, { size: 14, className: "text-muted-foreground" }),
            /* @__PURE__ */ s("div", { className: "flex items-center gap-1", children: [
              /* @__PURE__ */ e(
                "input",
                {
                  type: "number",
                  min: 0,
                  max: 23,
                  value: String(H).padStart(2, "0"),
                  onChange: (C) => U(Math.max(0, Math.min(23, Number(C.target.value)))),
                  className: "w-10 text-center bg-input-background border border-border rounded-[var(--radius-sm)] py-1 text-foreground focus:outline-none focus:ring-1 focus:ring-ring/30",
                  style: G
                }
              ),
              /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: G, children: ":" }),
              /* @__PURE__ */ e(
                "input",
                {
                  type: "number",
                  min: 0,
                  max: 59,
                  value: String(Z).padStart(2, "0"),
                  onChange: (C) => K(Math.max(0, Math.min(59, Number(C.target.value)))),
                  className: "w-10 text-center bg-input-background border border-border rounded-[var(--radius-sm)] py-1 text-foreground focus:outline-none focus:ring-1 focus:ring-ring/30",
                  style: G
                }
              )
            ] }),
            /* @__PURE__ */ e(
              "button",
              {
                onClick: Fe,
                className: "ml-auto px-3 py-1 rounded-[var(--radius-md)] bg-primary text-primary-foreground hover:bg-primary/90 transition-colors cursor-pointer",
                style: G,
                children: "Apply"
              }
            )
          ] }),
          /* @__PURE__ */ s("div", { className: "px-4 py-2.5 border-t border-border bg-card flex items-center justify-between", children: [
            y ? /* @__PURE__ */ e(
              "button",
              {
                onClick: xe,
                className: "text-primary hover:underline cursor-pointer",
                style: G,
                children: "Today"
              }
            ) : /* @__PURE__ */ e("span", {}),
            $ ?? /* @__PURE__ */ e(
              "button",
              {
                onClick: () => {
                  I(!1), F("days");
                },
                className: "text-muted-foreground hover:text-foreground cursor-pointer",
                style: G,
                children: "Close"
              }
            )
          ] })
        ]
      }
    )
  ] });
}
const Or = {
  sm: "max-w-sm",
  md: "max-w-lg",
  lg: "max-w-2xl",
  xl: "max-w-4xl",
  full: "max-w-[calc(100vw-2rem)] max-h-[calc(100vh-2rem)]"
};
function Ur({
  open: t,
  onClose: r,
  title: o,
  description: n,
  size: a = "md",
  children: i,
  footer: l,
  closeOnOverlay: c = !0,
  showCloseButton: d = !0,
  className: f = ""
}) {
  const m = X(null);
  return V(() => {
    if (!t) return;
    const v = (p) => {
      p.key === "Escape" && r();
    };
    return document.addEventListener("keydown", v), document.body.style.overflow = "hidden", () => {
      document.removeEventListener("keydown", v), document.body.style.overflow = "";
    };
  }, [t, r]), V(() => {
    t && m.current && m.current.focus();
  }, [t]), t ? /* @__PURE__ */ s("div", { className: "fixed inset-0 z-[100] flex items-center justify-center p-4", children: [
    /* @__PURE__ */ e(
      "div",
      {
        className: "absolute inset-0 bg-foreground/40 backdrop-blur-sm animate-in fade-in duration-200",
        onClick: c ? r : void 0
      }
    ),
    /* @__PURE__ */ s(
      "div",
      {
        ref: m,
        tabIndex: -1,
        role: "dialog",
        "aria-modal": "true",
        "aria-labelledby": o ? "modal-title" : void 0,
        className: `relative w-full ${Or[a]} bg-card rounded-[var(--radius-lg)] shadow-lg border border-border animate-in zoom-in-95 fade-in duration-200 flex flex-col ${a === "full" ? "h-full" : "max-h-[85vh]"} ${f}`,
        children: [
          (o || d) && /* @__PURE__ */ s("div", { className: "flex items-start justify-between gap-4 px-6 pt-6 pb-2 flex-shrink-0", children: [
            /* @__PURE__ */ s("div", { className: "min-w-0 flex-1", children: [
              o && /* @__PURE__ */ e("h4", { id: "modal-title", className: "text-foreground", children: o }),
              n && /* @__PURE__ */ e("span", { className: "text-muted-foreground block mt-0.5", style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" }, children: n })
            ] }),
            d && /* @__PURE__ */ e(
              "button",
              {
                onClick: r,
                className: "flex-shrink-0 w-8 h-8 rounded-[var(--radius-sm)] flex items-center justify-center text-muted-foreground hover:text-foreground hover:bg-muted transition-colors cursor-pointer",
                "aria-label": "Close",
                children: /* @__PURE__ */ e(J, { size: 16 })
              }
            )
          ] }),
          /* @__PURE__ */ e("div", { className: "flex-1 overflow-y-auto px-6 py-4", style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" }, children: i }),
          l && /* @__PURE__ */ e("div", { className: "flex items-center justify-end gap-2 px-6 py-4 border-t border-border flex-shrink-0", children: l })
        ]
      }
    )
  ] }) : null;
}
function Xo({
  open: t,
  onClose: r,
  onConfirm: o,
  title: n = "Are you sure?",
  description: a = "This action cannot be undone.",
  confirmLabel: i = "Confirm",
  cancelLabel: l = "Cancel",
  variant: c = "default"
}) {
  const d = { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)" };
  return /* @__PURE__ */ e(Ur, { open: t, onClose: r, title: n, description: a, size: "sm", children: /* @__PURE__ */ s("div", { className: "flex items-center justify-end gap-2 pt-4", children: [
    /* @__PURE__ */ e(
      "button",
      {
        onClick: r,
        className: "px-4 py-2 rounded-[var(--radius-md)] border border-border text-foreground hover:bg-muted transition-colors cursor-pointer",
        style: d,
        children: l
      }
    ),
    /* @__PURE__ */ e(
      "button",
      {
        onClick: () => {
          o(), r();
        },
        className: `px-4 py-2 rounded-[var(--radius-md)] text-primary-foreground transition-colors cursor-pointer ${c === "destructive" ? "bg-destructive hover:bg-destructive/90" : "bg-primary hover:bg-primary/90"}`,
        style: d,
        children: i
      }
    )
  ] }) });
}
const Kr = {
  sm: {
    btn: "min-w-[28px] h-7 px-1.5",
    icon: 14,
    text: {
      fontFamily: "var(--font-button)",
      fontSize: "var(--text-button)",
      fontWeight: "var(--weight-button)"
    },
    gap: "gap-1"
  },
  md: {
    btn: "min-w-[34px] h-9 px-2",
    icon: 16,
    text: {
      fontFamily: "var(--font-button)",
      fontSize: "var(--text-button)",
      fontWeight: "var(--weight-button)"
    },
    gap: "gap-1.5"
  },
  lg: {
    btn: "min-w-[42px] h-11 px-3",
    icon: 18,
    text: {
      fontFamily: "var(--font-button)",
      fontSize: "var(--text-label)",
      fontWeight: "var(--weight-button)"
    },
    gap: "gap-2"
  }
}, Xr = {
  default: {
    base: "bg-transparent text-foreground border border-transparent",
    active: "bg-primary text-primary-foreground border-primary shadow-elevation-sm",
    hover: "hover:bg-muted"
  },
  outlined: {
    base: "bg-transparent text-foreground border border-border",
    active: "bg-primary text-primary-foreground border-primary shadow-elevation-sm",
    hover: "hover:bg-muted hover:border-muted"
  },
  filled: {
    base: "bg-muted/50 text-foreground border border-transparent",
    active: "bg-primary text-primary-foreground border-primary shadow-elevation-sm",
    hover: "hover:bg-muted"
  },
  minimal: {
    base: "bg-transparent text-foreground border border-transparent",
    active: "bg-accent text-accent-foreground border-accent",
    hover: "hover:text-primary"
  }
};
function Gr(t, r, o) {
  return te(() => {
    if (o * 2 + 5 >= r)
      return Array.from({ length: r }, (f, m) => m + 1);
    const a = Math.max(t - o, 1), i = Math.min(t + o, r), l = a > 2, c = i < r - 1;
    if (!l && c) {
      const f = 3 + 2 * o;
      return [...Array.from({ length: f }, (v, p) => p + 1), "dots-right", r];
    }
    if (l && !c) {
      const f = 3 + 2 * o;
      return [1, "dots-left", ...Array.from(
        { length: f },
        (v, p) => r - f + p + 1
      )];
    }
    return [1, "dots-left", ...Array.from(
      { length: i - a + 1 },
      (f, m) => a + m
    ), "dots-right", r];
  }, [t, r, o]);
}
function Go({
  currentPage: t,
  totalPages: r,
  onPageChange: o,
  siblingCount: n = 1,
  showFirstLast: a = !1,
  showPrevNext: i = !0,
  showPageSize: l = !1,
  pageSizeOptions: c = [10, 20, 50, 100],
  pageSize: d = 10,
  onPageSizeChange: f,
  totalItems: m,
  size: v = "md",
  variant: p = "default",
  disabled: u = !1,
  showPageInfo: S = !1,
  showItemsInfo: N = !1,
  prevLabel: h,
  nextLabel: x
}) {
  const D = Gr(t, r, n), b = Kr[v], y = Xr[p], $ = P(
    (L) => {
      !u && L >= 1 && L <= r && o(L);
    },
    [u, r, o]
  ), k = (L, Y) => `inline-flex items-center justify-center rounded-[var(--radius-md)] transition-all cursor-pointer select-none
     ${b.btn}
     ${Y ? "opacity-40 pointer-events-none cursor-default" : ""}
     ${L ? y.active : `${y.base} ${Y ? "" : y.hover}`}`, M = m ? (t - 1) * d + 1 : 0, I = m ? Math.min(t * d, m) : 0, W = {
    fontFamily: "var(--font-label)",
    fontSize: "var(--text-label)",
    fontWeight: "var(--weight-label)"
  };
  return /* @__PURE__ */ s("div", { className: "flex flex-wrap items-center justify-between gap-4", children: [
    /* @__PURE__ */ s("div", { className: "flex items-center gap-4", children: [
      N && m !== void 0 && /* @__PURE__ */ s("span", { className: "text-muted-foreground", style: W, children: [
        M,
        "–",
        I,
        " of ",
        m
      ] }),
      l && /* @__PURE__ */ s("div", { className: "flex items-center gap-2", children: [
        /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: W, children: "Rows per page:" }),
        /* @__PURE__ */ e(
          "select",
          {
            value: d,
            onChange: (L) => f == null ? void 0 : f(Number(L.target.value)),
            disabled: u,
            className: "bg-input-background text-foreground border border-border rounded-[var(--radius-md)] px-2 py-1 cursor-pointer focus:outline-none focus:ring-2 focus:ring-ring/30",
            style: {
              fontFamily: "var(--font-button)",
              fontSize: "var(--text-button)",
              fontWeight: "var(--weight-button)"
            },
            children: c.map((L) => /* @__PURE__ */ e("option", { value: L, children: L }, L))
          }
        )
      ] })
    ] }),
    /* @__PURE__ */ s(
      "nav",
      {
        className: `flex items-center ${b.gap}`,
        "aria-label": "Pagination",
        role: "navigation",
        children: [
          a && /* @__PURE__ */ e(
            "button",
            {
              onClick: () => $(1),
              disabled: u || t <= 1,
              className: k(!1, u || t <= 1),
              style: b.text,
              "aria-label": "First page",
              title: "First page",
              children: /* @__PURE__ */ e(tt, { size: b.icon })
            }
          ),
          i && /* @__PURE__ */ e(
            "button",
            {
              onClick: () => $(t - 1),
              disabled: u || t <= 1,
              className: k(!1, u || t <= 1),
              style: b.text,
              "aria-label": "Previous page",
              children: h ?? /* @__PURE__ */ e($e, { size: b.icon })
            }
          ),
          D.map((L, Y) => L === "dots-left" || L === "dots-right" ? /* @__PURE__ */ e(
            "span",
            {
              className: `inline-flex items-center justify-center text-muted-foreground ${b.btn}`,
              style: b.text,
              children: /* @__PURE__ */ e(Lt, { size: b.icon })
            },
            L
          ) : /* @__PURE__ */ e(
            "button",
            {
              onClick: () => $(L),
              disabled: u,
              className: k(L === t, u),
              style: b.text,
              "aria-label": `Page ${L}`,
              "aria-current": L === t ? "page" : void 0,
              children: L
            },
            L
          )),
          i && /* @__PURE__ */ e(
            "button",
            {
              onClick: () => $(t + 1),
              disabled: u || t >= r,
              className: k(!1, u || t >= r),
              style: b.text,
              "aria-label": "Next page",
              children: x ?? /* @__PURE__ */ e(le, { size: b.icon })
            }
          ),
          a && /* @__PURE__ */ e(
            "button",
            {
              onClick: () => $(r),
              disabled: u || t >= r,
              className: k(!1, u || t >= r),
              style: b.text,
              "aria-label": "Last page",
              title: "Last page",
              children: /* @__PURE__ */ e(rt, { size: b.icon })
            }
          )
        ]
      }
    ),
    S && /* @__PURE__ */ s("span", { className: "text-muted-foreground", style: W, children: [
      "Page ",
      t,
      " of ",
      r
    ] })
  ] });
}
const qr = {
  sm: { input: "h-8 px-8", icon: 14, style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" } },
  md: { input: "h-10 px-9", icon: 16, style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" } },
  lg: { input: "h-12 px-10", icon: 18, style: { fontFamily: "var(--font-caption)", fontSize: "var(--text-caption)", fontWeight: "var(--weight-caption)" } }
}, Jr = {
  default: "bg-card border border-border focus-within:border-ring focus-within:ring-2 focus-within:ring-ring/20",
  outlined: "bg-transparent border-2 border-border focus-within:border-primary",
  filled: "bg-muted/50 border border-transparent focus-within:border-ring focus-within:bg-card"
};
function qo({
  value: t,
  onChange: r,
  onSearch: o,
  placeholder: n = "Search...",
  size: a = "md",
  variant: i = "default",
  loading: l = !1,
  suggestions: c = [],
  onSuggestionSelect: d,
  clearable: f = !0,
  disabled: m = !1,
  autoFocus: v = !1,
  debounce: p = 0,
  className: u = ""
}) {
  const [S, N] = z(""), [h, x] = z(!1), [D, b] = z(-1), y = t ?? S, $ = X(null), k = X(null), M = X(void 0), I = qr[a], W = P((w) => {
    t === void 0 && N(w), r == null || r(w), p > 0 && (clearTimeout(M.current), M.current = setTimeout(() => o == null ? void 0 : o(w), p));
  }, [t, r, o, p]), L = (w) => {
    w.key === "Enter" ? D >= 0 && c[D] ? (d == null || d(c[D]), x(!1)) : o == null || o(y) : w.key === "ArrowDown" ? (w.preventDefault(), b((T) => Math.min(T + 1, c.length - 1))) : w.key === "ArrowUp" ? (w.preventDefault(), b((T) => Math.max(T - 1, -1))) : w.key === "Escape" && x(!1);
  };
  V(() => {
    const w = (T) => {
      k.current && !k.current.contains(T.target) && x(!1);
    };
    return document.addEventListener("mousedown", w), () => document.removeEventListener("mousedown", w);
  }, []), V(() => {
    x(c.length > 0 && y.length > 0), b(-1);
  }, [c, y]);
  const Y = { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" }, F = { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" };
  return /* @__PURE__ */ s("div", { ref: k, className: `relative ${u}`, children: [
    /* @__PURE__ */ s("div", { className: `relative rounded-[var(--radius-md)] transition-all ${Jr[i]} ${m ? "opacity-50 pointer-events-none" : ""}`, children: [
      /* @__PURE__ */ e("span", { className: "absolute left-3 top-1/2 -translate-y-1/2 text-muted-foreground pointer-events-none", children: /* @__PURE__ */ e(Be, { size: I.icon }) }),
      /* @__PURE__ */ e(
        "input",
        {
          ref: $,
          type: "text",
          value: y,
          onChange: (w) => W(w.target.value),
          onKeyDown: L,
          onFocus: () => c.length > 0 && y.length > 0 && x(!0),
          placeholder: n,
          disabled: m,
          autoFocus: v,
          className: `w-full ${I.input} rounded-[var(--radius-md)] bg-transparent outline-none text-foreground placeholder:text-muted-foreground`,
          style: I.style
        }
      ),
      /* @__PURE__ */ s("span", { className: "absolute right-3 top-1/2 -translate-y-1/2 flex items-center gap-1", children: [
        l && /* @__PURE__ */ e(Se, { size: I.icon, className: "animate-spin text-muted-foreground" }),
        f && y && !l && /* @__PURE__ */ e(
          "button",
          {
            onClick: () => {
              var w;
              W(""), (w = $.current) == null || w.focus();
            },
            className: "text-muted-foreground hover:text-foreground transition-colors cursor-pointer",
            children: /* @__PURE__ */ e(J, { size: I.icon })
          }
        )
      ] })
    ] }),
    h && c.length > 0 && /* @__PURE__ */ e("div", { className: "absolute top-full left-0 right-0 mt-1 bg-popover border border-border rounded-[var(--radius-md)] shadow-lg z-50 overflow-hidden max-h-64 overflow-y-auto", style: { padding: "var(--Spacing--Spacing-sm)" }, children: c.map((w, T) => /* @__PURE__ */ s(
      "button",
      {
        onClick: () => {
          d == null || d(w), x(!1);
        },
        className: `w-full text-left flex items-center rounded-[var(--radius-sm)] transition-colors cursor-pointer ${T === D ? "bg-[var(--Colors--Background--bg-brand-secondary)]" : "hover:bg-[var(--Colors--Background--bg-primary_hover)]"}`,
        style: { padding: "var(--Spacing--Spacing-md) var(--Spacing--Spacing-2xl)", gap: "var(--Spacing--Spacing-lg)" },
        children: [
          w.icon && /* @__PURE__ */ e("span", { className: "text-muted-foreground flex-shrink-0", children: w.icon }),
          /* @__PURE__ */ s("div", { className: "min-w-0 flex-1", children: [
            /* @__PURE__ */ e("span", { className: "text-foreground block truncate", style: Y, children: w.label }),
            w.description && /* @__PURE__ */ e("span", { className: "text-muted-foreground block truncate", style: { marginTop: "var(--Spacing--Spacing-xxs)", ...F }, children: w.description })
          ] })
        ]
      },
      w.id
    )) })
  ] });
}
const pe = {
  sm: "px-3 py-2",
  md: "px-4 py-3",
  lg: "px-5 py-4"
};
function Jo({
  columns: t,
  data: r,
  size: o = "md",
  striped: n = !1,
  hoverable: a = !0,
  bordered: i = !1,
  selectable: l = !1,
  selectedRows: c,
  onSelectionChange: d,
  loading: f = !1,
  emptyMessage: m = "No data available",
  stickyHeader: v = !1,
  className: p = ""
}) {
  const [u, S] = z(null), [N, h] = z("asc"), [x, D] = z(/* @__PURE__ */ new Set()), b = c ?? x, y = (w) => {
    c || D(w), d == null || d(w);
  }, $ = te(() => u ? [...r].sort((w, T) => {
    const H = w[u], U = T[u];
    if (H == null) return 1;
    if (U == null) return -1;
    const Z = typeof H == "string" ? H.localeCompare(U) : H - U;
    return N === "asc" ? Z : -Z;
  }) : r, [r, u, N]), k = (w) => {
    u === w ? h((T) => T === "asc" ? "desc" : "asc") : (S(w), h("asc"));
  }, M = r.length > 0 && b.size === r.length, I = b.size > 0 && !M, W = () => {
    y(M ? /* @__PURE__ */ new Set() : new Set(r.map((w, T) => T)));
  }, L = (w) => {
    const T = new Set(b);
    T.has(w) ? T.delete(w) : T.add(w), y(T);
  }, Y = { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" }, F = { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)" };
  return /* @__PURE__ */ e("div", { className: `rounded-[var(--radius-lg)] border border-border overflow-hidden ${p}`, children: /* @__PURE__ */ e("div", { className: "overflow-x-auto", children: /* @__PURE__ */ s("table", { className: "w-full border-collapse", children: [
    /* @__PURE__ */ e("thead", { className: v ? "sticky top-0 z-10" : "", children: /* @__PURE__ */ s("tr", { className: "bg-muted/50 border-b border-border", children: [
      l && /* @__PURE__ */ e("th", { className: `${pe[o]} w-12`, children: /* @__PURE__ */ e(
        "input",
        {
          type: "checkbox",
          checked: M,
          ref: (w) => {
            w && (w.indeterminate = I);
          },
          onChange: W,
          className: "w-4 h-4 rounded-[var(--radius-sm)] border-border accent-primary cursor-pointer"
        }
      ) }),
      t.map((w) => /* @__PURE__ */ e(
        "th",
        {
          className: `${pe[o]} text-${w.align ?? "left"} text-muted-foreground ${w.sortable ? "cursor-pointer select-none hover:text-foreground transition-colors" : ""} ${i ? "border-x border-border first:border-l-0 last:border-r-0" : ""}`,
          style: { ...F, width: w.width },
          onClick: () => w.sortable && k(w.key),
          children: /* @__PURE__ */ s("span", { className: "inline-flex items-center gap-1", children: [
            w.header,
            w.sortable && /* @__PURE__ */ e("span", { className: "text-muted-foreground/60", children: u === w.key ? N === "asc" ? /* @__PURE__ */ e(sr, { size: 14 }) : /* @__PURE__ */ e(Ye, { size: 14 }) : /* @__PURE__ */ e(lr, { size: 12 }) })
          ] })
        },
        w.key
      ))
    ] }) }),
    /* @__PURE__ */ e("tbody", { children: f ? /* @__PURE__ */ e("tr", { children: /* @__PURE__ */ e("td", { colSpan: t.length + (l ? 1 : 0), className: `${pe[o]} text-center text-muted-foreground`, children: /* @__PURE__ */ s("div", { className: "flex items-center justify-center gap-2 py-8", children: [
      /* @__PURE__ */ e("div", { className: "w-5 h-5 border-2 border-primary border-t-transparent rounded-full animate-spin" }),
      /* @__PURE__ */ e("span", { style: Y, children: "Loading..." })
    ] }) }) }) : $.length === 0 ? /* @__PURE__ */ e("tr", { children: /* @__PURE__ */ e("td", { colSpan: t.length + (l ? 1 : 0), className: `${pe[o]} text-center text-muted-foreground py-12`, children: /* @__PURE__ */ e("span", { style: Y, children: m }) }) }) : $.map((w, T) => {
      const H = b.has(T);
      return /* @__PURE__ */ s(
        "tr",
        {
          className: `border-b border-border last:border-b-0 transition-colors ${H ? "bg-primary/5" : n && T % 2 === 1 ? "bg-muted/20" : "bg-card"} ${a ? "hover:bg-[var(--Colors--Background--bg-primary_hover)]" : ""}`,
          children: [
            l && /* @__PURE__ */ e("td", { className: `${pe[o]} w-12`, children: /* @__PURE__ */ e(
              "input",
              {
                type: "checkbox",
                checked: H,
                onChange: () => L(T),
                className: "w-4 h-4 rounded-[var(--radius-sm)] border-border accent-primary cursor-pointer"
              }
            ) }),
            t.map((U) => /* @__PURE__ */ e(
              "td",
              {
                className: `${pe[o]} text-${U.align ?? "left"} text-foreground ${i ? "border-x border-border first:border-l-0 last:border-r-0" : ""}`,
                style: Y,
                children: U.render ? U.render(w[U.key], w, T) : w[U.key]
              },
              U.key
            ))
          ]
        },
        T
      );
    }) })
  ] }) }) });
}
const Zr = {
  sm: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)", padding: "6px 12px" },
  md: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)", padding: "8px 16px" },
  lg: { fontFamily: "var(--font-button)", fontSize: "var(--text-label)", fontWeight: "var(--weight-button)", padding: "10px 20px" }
};
function Zo({ tabs: t, variant: r = "default", size: o = "md", defaultTab: n, activeTab: a, onChange: i, fullWidth: l = !1, className: c = "" }) {
  var D, b;
  const [d, f] = z(n ?? ((D = t[0]) == null ? void 0 : D.id) ?? ""), m = a ?? d, v = X(null), [p, u] = z({ left: 0, width: 0 }), S = (y) => {
    a || f(y), i == null || i(y);
  };
  V(() => {
    if (r !== "underline" && r !== "default") return;
    const y = v.current;
    if (!y) return;
    const $ = y.querySelector(`[data-tab-id="${m}"]`);
    $ && u({ left: $.offsetLeft, width: $.offsetWidth });
  }, [m, r]);
  const N = (b = t.find((y) => y.id === m)) == null ? void 0 : b.content, h = (() => {
    switch (r) {
      case "bordered":
        return "border border-border rounded-[var(--radius)] bg-muted/30 p-1 gap-1";
      case "pills":
        return "gap-1.5";
      case "underline":
        return "border-b border-border relative";
      default:
        return "border-b border-border relative";
    }
  })(), x = (y, $) => {
    const k = "relative flex items-center gap-1.5 transition-all cursor-pointer outline-none focus-visible:ring-2 focus-visible:ring-ring whitespace-nowrap";
    if ($) return `${k} opacity-40 !cursor-not-allowed`;
    switch (r) {
      case "bordered":
        return `${k} rounded-[var(--radius-sm)] ${y ? "bg-card text-foreground shadow-elevation-sm" : "text-muted-foreground hover:text-foreground hover:bg-card/50"}`;
      case "pills":
        return `${k} rounded-[var(--radius)] ${y ? "bg-primary text-primary-foreground" : "text-muted-foreground hover:text-foreground hover:bg-muted"}`;
      case "underline":
        return `${k} ${y ? "text-primary" : "text-muted-foreground hover:text-foreground"}`;
      default:
        return `${k} ${y ? "text-primary" : "text-muted-foreground hover:text-foreground"}`;
    }
  };
  return /* @__PURE__ */ s("div", { className: c, children: [
    /* @__PURE__ */ s("div", { ref: v, className: `flex ${l ? "" : "inline-flex"} ${h}`, role: "tablist", children: [
      (r === "default" || r === "underline") && /* @__PURE__ */ e(
        "div",
        {
          className: "absolute bottom-0 h-0.5 bg-primary transition-all duration-200 rounded-full",
          style: { left: p.left, width: p.width }
        }
      ),
      t.map((y) => {
        const $ = m === y.id;
        return /* @__PURE__ */ s(
          "button",
          {
            "data-tab-id": y.id,
            role: "tab",
            "aria-selected": $,
            disabled: y.disabled,
            onClick: () => !y.disabled && S(y.id),
            className: `${x($, !!y.disabled)} ${l ? "flex-1 justify-center" : ""}`,
            style: Zr[o],
            children: [
              y.icon && /* @__PURE__ */ e("span", { className: "flex-shrink-0", children: y.icon }),
              /* @__PURE__ */ e("span", { children: y.label }),
              y.badge !== void 0 && /* @__PURE__ */ e(
                "span",
                {
                  className: `ml-1 px-1.5 py-0.5 rounded-full min-w-[18px] text-center ${$ && r === "pills" ? "bg-primary-foreground/20 text-primary-foreground" : "bg-muted text-muted-foreground"}`,
                  style: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)", lineHeight: "1" },
                  children: y.badge
                }
              )
            ]
          },
          y.id
        );
      })
    ] }),
    N && /* @__PURE__ */ e("div", { className: "pt-4", role: "tabpanel", children: N })
  ] });
}
const Qr = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, Pr = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
};
function Ht({ title: t, children: r, open: o, onToggle: n, icon: a }) {
  return /* @__PURE__ */ s("div", { className: "border-b border-border last:border-b-0", children: [
    /* @__PURE__ */ s(
      "button",
      {
        onClick: n,
        className: "w-full flex items-center gap-3 px-4 py-3 text-left hover:bg-muted/20 transition-colors cursor-pointer",
        children: [
          a && /* @__PURE__ */ e("span", { className: "text-primary flex-shrink-0", children: a }),
          /* @__PURE__ */ e("span", { className: "flex-1 text-foreground", style: Qr, children: t }),
          /* @__PURE__ */ e(
            Ye,
            {
              size: 16,
              className: `text-muted-foreground transition-transform ${o ? "rotate-180" : ""}`
            }
          )
        ]
      }
    ),
    /* @__PURE__ */ e("div", { className: `overflow-hidden transition-all ${o ? "max-h-96 opacity-100" : "max-h-0 opacity-0"}`, children: /* @__PURE__ */ e("div", { className: "px-4 pb-4 text-muted-foreground", style: Pr, children: r }) })
  ] });
}
function Vr({
  type: t = "single",
  items: r,
  defaultOpen: o,
  value: n,
  onChange: a,
  className: i = ""
}) {
  const [l, c] = z(
    typeof o == "string" ? o : null
  ), [d, f] = z(
    Array.isArray(o) ? new Set(o) : /* @__PURE__ */ new Set()
  ), m = n !== void 0, v = (u) => m ? t === "single" ? n === u : Array.isArray(n) && n.includes(u) : t === "single" ? l === u : d.has(u), p = (u) => {
    if (t === "single") {
      const S = (m ? n === u : l === u) ? null : u;
      m || c(S), a == null || a(S);
    } else if (m) {
      const S = Array.isArray(n) ? n : [], N = S.includes(u) ? S.filter((h) => h !== u) : [...S, u];
      a == null || a(N);
    } else
      f((S) => {
        const N = new Set(S);
        return N.has(u) ? N.delete(u) : N.add(u), N;
      });
  };
  return /* @__PURE__ */ e("div", { className: `rounded-[var(--radius)] border border-border overflow-hidden ${i}`, children: r.map((u) => /* @__PURE__ */ e(
    Ht,
    {
      title: u.title,
      open: v(u.id),
      onToggle: () => p(u.id),
      icon: u.icon,
      children: u.content
    },
    u.id
  )) });
}
Vr.displayName = "Accordion";
Ht.displayName = "AccordionItem";
const ht = ["bg-primary", "bg-chart-2", "bg-chart-5", "bg-destructive", "bg-secondary"], en = {
  xs: { box: "w-6 h-6", font: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)" } },
  sm: { box: "w-8 h-8", font: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)" } },
  md: { box: "w-10 h-10", font: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)" } },
  lg: { box: "w-14 h-14", font: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)" } },
  xl: { box: "w-20 h-20", font: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-button)" } }
}, tn = {
  xs: "w-2 h-2",
  sm: "w-2.5 h-2.5",
  md: "w-3 h-3",
  lg: "w-3.5 h-3.5",
  xl: "w-4 h-4"
}, rn = {
  online: "bg-chart-2",
  offline: "bg-secondary",
  busy: "bg-destructive",
  away: "bg-chart-5"
}, nn = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
};
function on({ src: t, name: r, size: o = "md", status: n, className: a = "" }) {
  const i = en[o], l = r ? r.split(" ").map((d) => d[0]).join("").slice(0, 2).toUpperCase() : "?", c = r ? ht[r.length % ht.length] : "bg-muted";
  return /* @__PURE__ */ s("div", { className: `relative inline-flex ${a}`, children: [
    t ? /* @__PURE__ */ e(
      "img",
      {
        src: t,
        alt: r || "avatar",
        className: `${i.box} rounded-full object-cover border-2 border-card`
      }
    ) : /* @__PURE__ */ e(
      "div",
      {
        className: `${i.box} rounded-full ${c} text-white flex items-center justify-center border-2 border-card`,
        style: i.font,
        children: l
      }
    ),
    n && /* @__PURE__ */ e(
      "span",
      {
        className: `absolute bottom-0 right-0 ${tn[o]} ${rn[n]} rounded-full border-2 border-card`
      }
    )
  ] });
}
function an({ children: t, max: r, className: o = "" }) {
  const n = et.Children.toArray(t), a = r ? n.slice(0, r) : n, i = r && n.length > r ? n.length - r : 0;
  return /* @__PURE__ */ s("div", { className: `flex -space-x-2 ${o}`, children: [
    a.map((l, c) => /* @__PURE__ */ e("div", { className: "relative", style: { zIndex: a.length - c }, children: l }, c)),
    i > 0 && /* @__PURE__ */ s(
      "div",
      {
        className: "relative w-10 h-10 rounded-full bg-muted text-foreground flex items-center justify-center border-2 border-card",
        style: { ...nn, zIndex: 0 },
        children: [
          "+",
          i
        ]
      }
    )
  ] });
}
on.displayName = "Avatar";
an.displayName = "AvatarGroup";
const sn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, ln = {
  sm: {
    fontFamily: "var(--font-button)",
    fontSize: "var(--text-button)",
    fontWeight: "var(--weight-label)"
  },
  md: sn,
  lg: {
    fontFamily: "var(--font-label)",
    fontSize: "var(--text-label)",
    fontWeight: "var(--weight-label)"
  }
};
function Qo({
  items: t,
  separator: r = "chevron",
  size: o = "md",
  maxItems: n
}) {
  const a = r === "slash" ? /* @__PURE__ */ e(ir, { size: 12, className: "text-muted-foreground" }) : r === "dot" ? /* @__PURE__ */ e(cr, { size: 16, className: "text-muted-foreground" }) : /* @__PURE__ */ e(le, { size: 12, className: "text-muted-foreground" }), i = o === "sm" ? "gap-1" : o === "lg" ? "gap-2.5" : "gap-1.5", l = ln[o];
  let c = t;
  if (n && t.length > n) {
    const d = t.slice(0, 1), f = t.slice(-(n - 1));
    c = [...d, { label: "…" }, ...f];
  }
  return /* @__PURE__ */ e("nav", { "aria-label": "Breadcrumb", children: /* @__PURE__ */ e("ol", { className: `flex items-center flex-wrap ${i}`, children: c.map((d, f) => /* @__PURE__ */ s("li", { className: "flex items-center gap-1.5", children: [
    f > 0 && /* @__PURE__ */ e("span", { className: "mx-0.5", children: a }),
    d.label === "…" ? /* @__PURE__ */ e("span", { className: "w-6 h-6 flex items-center justify-center rounded-[var(--radius-sm)] text-muted-foreground hover:bg-accent cursor-pointer", children: /* @__PURE__ */ e(Lt, { size: 14 }) }) : f === c.length - 1 ? /* @__PURE__ */ s(
      "span",
      {
        className: "inline-flex items-center gap-1.5 text-foreground",
        style: {
          ...l,
          fontWeight: "var(--weight-button)"
        },
        children: [
          d.icon,
          d.label
        ]
      }
    ) : /* @__PURE__ */ s(
      "a",
      {
        href: d.href || "#",
        onClick: (m) => m.preventDefault(),
        className: "inline-flex items-center gap-1.5 text-muted-foreground hover:text-foreground transition-colors",
        style: l,
        children: [
          d.icon,
          d.label
        ]
      }
    )
  ] }, f)) }) });
}
const cn = {
  none: "",
  sm: "shadow-sm",
  md: "shadow-md",
  lg: "shadow-lg"
};
function dn({ children: t, className: r, hover: o, elevation: n = "none" }) {
  return /* @__PURE__ */ e(
    "div",
    {
      className: `rounded-[var(--radius-lg)] border border-border bg-card overflow-hidden ${cn[n] ?? ""} ${o ? "hover:border-primary/40 hover:shadow-md transition-all cursor-pointer" : ""} ${r ?? ""}`,
      children: t
    }
  );
}
dn.displayName = "Card";
function un({ children: t, action: r }) {
  return /* @__PURE__ */ s("div", { className: "px-5 py-4 border-b border-border flex items-center justify-between", children: [
    /* @__PURE__ */ e("div", { children: t }),
    r
  ] });
}
un.displayName = "CardHeader";
function fn({ children: t }) {
  return /* @__PURE__ */ e("div", { className: "px-5 py-4", children: t });
}
fn.displayName = "CardBody";
function mn({ children: t }) {
  return /* @__PURE__ */ e("div", { className: "px-5 py-3 border-t border-border bg-muted/10 flex items-center gap-2", children: t });
}
mn.displayName = "CardFooter";
const hn = [
  "#ef4444",
  "#f97316",
  "#eab308",
  "#22c55e",
  "#06b6d4",
  "#3b82f6",
  "#8b5cf6",
  "#ec4899",
  "#991b1b",
  "#9a3412",
  "#854d0e",
  "#166534",
  "#155e75",
  "#1e40af",
  "#5b21b6",
  "#9d174d",
  "#fca5a5",
  "#fdba74",
  "#fde047",
  "#86efac",
  "#67e8f9",
  "#93c5fd",
  "#c4b5fd",
  "#f9a8d4"
];
function bn(t) {
  const r = parseInt(t.slice(1, 3), 16), o = parseInt(t.slice(3, 5), 16), n = parseInt(t.slice(5, 7), 16);
  return `rgb(${r}, ${o}, ${n})`;
}
function gn(t) {
  const r = parseInt(t.slice(1, 3), 16) / 255, o = parseInt(t.slice(3, 5), 16) / 255, n = parseInt(t.slice(5, 7), 16) / 255, a = Math.max(r, o, n), i = Math.min(r, o, n), l = (a + i) / 2;
  if (a === i) return `hsl(0, 0%, ${Math.round(l * 100)}%)`;
  const c = a - i, d = l > 0.5 ? c / (2 - a - i) : c / (a + i);
  let f = 0;
  return a === r ? f = ((o - n) / c + (o < n ? 6 : 0)) / 6 : a === o ? f = ((n - r) / c + 2) / 6 : f = ((r - o) / c + 4) / 6, `hsl(${Math.round(f * 360)}, ${Math.round(d * 100)}%, ${Math.round(l * 100)}%)`;
}
const pn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, vn = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, bt = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
};
function Po({
  value: t,
  onChange: r,
  label: o,
  presets: n,
  showInput: a = !0,
  showFormats: i,
  size: l = "md"
}) {
  const [c, d] = z(!1), f = X(null), m = l === "sm" ? "w-5 h-5" : l === "lg" ? "w-8 h-8" : "w-6 h-6", v = n || hn, p = (u) => {
    navigator.clipboard.writeText(u).then(() => {
      d(!0), setTimeout(() => d(!1), 1500);
    });
  };
  return /* @__PURE__ */ s("div", { className: "space-y-3", children: [
    o && /* @__PURE__ */ e("label", { className: "block text-foreground", style: pn, children: o }),
    /* @__PURE__ */ s("div", { className: "flex items-center gap-3", children: [
      /* @__PURE__ */ s("div", { className: "relative", children: [
        /* @__PURE__ */ e(
          "div",
          {
            className: "w-10 h-10 rounded-[var(--radius-md)] border border-border cursor-pointer overflow-hidden",
            style: { backgroundColor: t },
            onClick: () => {
              var u;
              return (u = f.current) == null ? void 0 : u.click();
            }
          }
        ),
        /* @__PURE__ */ e(
          "input",
          {
            ref: f,
            type: "color",
            value: t,
            onChange: (u) => r(u.target.value),
            className: "absolute inset-0 opacity-0 cursor-pointer w-full h-full"
          }
        )
      ] }),
      a && /* @__PURE__ */ s("div", { className: "flex items-center gap-1.5", children: [
        /* @__PURE__ */ e(
          "input",
          {
            value: t,
            onChange: (u) => {
              (/^#[0-9a-fA-F]{0,6}$/.test(u.target.value) || u.target.value === "#") && r(u.target.value);
            },
            className: "w-24 px-2.5 py-1.5 rounded-[var(--radius-md)] border border-border bg-background text-foreground outline-none focus:border-primary",
            style: bt,
            maxLength: 7
          }
        ),
        /* @__PURE__ */ e(
          "button",
          {
            onClick: () => p(t),
            className: "w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] text-muted-foreground hover:text-foreground hover:bg-accent transition-colors cursor-pointer",
            children: c ? /* @__PURE__ */ e(fe, { size: 14, className: "text-chart-2" }) : /* @__PURE__ */ e(it, { size: 14 })
          }
        )
      ] })
    ] }),
    v.length > 0 && /* @__PURE__ */ e("div", { className: "flex flex-wrap gap-1.5", children: v.map((u) => /* @__PURE__ */ e(
      "button",
      {
        onClick: () => r(u),
        className: `${m} rounded-[var(--radius-sm)] border-2 transition-transform hover:scale-110 cursor-pointer
                ${t === u ? "border-foreground ring-1 ring-foreground/20" : "border-transparent"}`,
        style: { backgroundColor: u },
        title: u
      },
      u
    )) }),
    i && /* @__PURE__ */ e("div", { className: "space-y-1", children: [
      { label: "HEX", val: t.toUpperCase() },
      { label: "RGB", val: bn(t) },
      { label: "HSL", val: gn(t) }
    ].map((u) => /* @__PURE__ */ s("div", { className: "flex items-center gap-2", children: [
      /* @__PURE__ */ e("span", { className: "w-8 text-muted-foreground", style: vn, children: u.label }),
      /* @__PURE__ */ e("code", { className: "text-foreground", style: bt, children: u.val }),
      /* @__PURE__ */ e(
        "button",
        {
          onClick: () => p(u.val),
          className: "text-muted-foreground hover:text-foreground cursor-pointer",
          children: /* @__PURE__ */ e(it, { size: 12 })
        }
      )
    ] }, u.label)) })
  ] });
}
const xn = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
};
function yn({
  label: t,
  orientation: r = "horizontal",
  dashed: o,
  spacing: n = "md",
  className: a = ""
}) {
  const i = { sm: "my-2", md: "my-4", lg: "my-8" }, l = o ? "border-dashed" : "border-solid";
  return r === "vertical" ? /* @__PURE__ */ e(
    "div",
    {
      className: `inline-block h-full min-h-[24px] border-l border-border ${l} mx-3 ${a}`
    }
  ) : t ? /* @__PURE__ */ s("div", { className: `flex items-center gap-3 ${i[n]} ${a}`, children: [
    /* @__PURE__ */ e("div", { className: `flex-1 border-t border-border ${l}` }),
    /* @__PURE__ */ e("span", { className: "text-muted-foreground flex-shrink-0", style: xn, children: t }),
    /* @__PURE__ */ e("div", { className: `flex-1 border-t border-border ${l}` })
  ] }) : /* @__PURE__ */ e("hr", { className: `border-t border-border ${l} ${i[n]} ${a}` });
}
yn.displayName = "Divider";
const wn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
};
function Nn({
  open: t,
  onClose: r,
  title: o,
  children: n,
  side: a = "right",
  size: i = "md",
  footer: l,
  className: c = ""
}) {
  if (!t) return null;
  const d = { sm: "w-72", md: "w-96", lg: "w-[480px]" }, f = { sm: "h-48", md: "h-72", lg: "h-96" }, m = {
    right: `top-0 right-0 h-full ${d[i]} animate-[slideRight_0.2s_ease]`,
    left: `top-0 left-0 h-full ${d[i]} animate-[slideLeft_0.2s_ease]`,
    top: `top-0 left-0 w-full ${f[i]} animate-[slideTop_0.2s_ease]`,
    bottom: `bottom-0 left-0 w-full ${f[i]} animate-[slideBottom_0.2s_ease]`
  };
  return /* @__PURE__ */ s("div", { className: "fixed inset-0 z-50", children: [
    /* @__PURE__ */ e("div", { className: "absolute inset-0 bg-foreground/20", onClick: r }),
    /* @__PURE__ */ s(
      "div",
      {
        className: `absolute bg-card border border-border shadow-lg flex flex-col ${m[a]} ${c}`,
        children: [
          o && /* @__PURE__ */ s("div", { className: "px-5 py-4 border-b border-border flex items-center justify-between flex-shrink-0", children: [
            /* @__PURE__ */ e("span", { className: "text-foreground", style: wn, children: o }),
            /* @__PURE__ */ e(
              "button",
              {
                onClick: r,
                className: "text-muted-foreground hover:text-foreground cursor-pointer",
                children: /* @__PURE__ */ e(J, { size: 18 })
              }
            )
          ] }),
          /* @__PURE__ */ e("div", { className: "flex-1 overflow-y-auto p-5", children: n }),
          l && /* @__PURE__ */ e("div", { className: "px-5 py-3 border-t border-border flex items-center gap-2 flex-shrink-0", children: l })
        ]
      }
    ),
    /* @__PURE__ */ e("style", { children: `
        @keyframes slideRight { from { transform: translateX(100%); } to { transform: translateX(0); } }
        @keyframes slideLeft { from { transform: translateX(-100%); } to { transform: translateX(0); } }
        @keyframes slideTop { from { transform: translateY(-100%); } to { transform: translateY(0); } }
        @keyframes slideBottom { from { transform: translateY(100%); } to { transform: translateY(0); } }
      ` })
  ] });
}
Nn.displayName = "Drawer";
const Sn = {
  sm: {
    trigger: "h-9 px-[var(--Spacing--Spacing-xl)] gap-[var(--Spacing--Spacing-sm)]",
    option: "rounded-[var(--radius-sm)]",
    optionStyle: { padding: "var(--Spacing--Spacing-sm) var(--Spacing--Spacing-xl)" },
    textStyle: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
    descStyle: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" },
    icon: 14,
    avatarSize: 20
  },
  md: {
    trigger: "h-11 px-[var(--Spacing--Spacing-2xl)] gap-[var(--Spacing--Spacing-lg)]",
    option: "rounded-[var(--radius-sm)]",
    optionStyle: { padding: "var(--Spacing--Spacing-md) var(--Spacing--Spacing-2xl)" },
    textStyle: { fontFamily: "var(--font-caption)", fontSize: "var(--text-caption)", fontWeight: "var(--weight-caption)" },
    descStyle: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
    icon: 16,
    avatarSize: 24
  },
  lg: {
    trigger: "h-13 px-[var(--Spacing--Spacing-3xl)] gap-[var(--Spacing--Spacing-xl)]",
    option: "rounded-[var(--radius-sm)]",
    optionStyle: { padding: "var(--Spacing--Spacing-lg) var(--Spacing--Spacing-3xl)" },
    textStyle: { fontFamily: "var(--font-p)", fontSize: "var(--text-p)", fontWeight: "var(--weight-p)" },
    descStyle: { fontFamily: "var(--font-caption)", fontSize: "var(--text-caption)", fontWeight: "var(--weight-caption)" },
    icon: 18,
    avatarSize: 28
  }
}, $n = {
  default: {
    base: "border border-border bg-input-background",
    hover: "hover:border-foreground/30",
    focus: "ring-2 ring-ring/20 border-ring"
  },
  outlined: {
    base: "border-2 border-border bg-transparent",
    hover: "hover:border-foreground/40",
    focus: "ring-2 ring-ring/20 border-ring"
  },
  filled: {
    base: "border border-transparent bg-muted",
    hover: "hover:bg-muted/80",
    focus: "ring-2 ring-ring/20 border-ring bg-input-background"
  },
  ghost: {
    base: "border border-transparent bg-transparent",
    hover: "hover:bg-muted/50",
    focus: "ring-2 ring-ring/20 bg-input-background"
  }
}, zn = {
  default: "",
  error: "!border-destructive ring-destructive/20",
  success: "!border-chart-2 ring-chart-2/20",
  warning: "!border-chart-5 ring-chart-5/20"
}, kn = ze(
  ({
    options: t,
    value: r,
    defaultValue: o,
    onChange: n,
    placeholder: a = "Select an option…",
    label: i,
    helperText: l,
    errorMessage: c,
    successMessage: d,
    size: f = "md",
    variant: m = "default",
    state: v = "default",
    disabled: p = !1,
    loading: u = !1,
    searchable: S = !1,
    clearable: N = !1,
    multiple: h = !1,
    maxSelections: x,
    grouped: D = !1,
    required: b = !1,
    id: y,
    className: $ = "",
    showSelectAll: k = !1,
    tagLimit: M,
    creatable: I = !1,
    onCreateOption: W,
    createLabel: L = 'Create "{query}"',
    emptyState: Y,
    footer: F,
    onSearch: w,
    searchDebounce: T = 300,
    searchLoading: H = !1,
    renderOption: U
  }, Z) => {
    const [K, Q] = z(!1), [O, ne] = z(""), [oe, ae] = z(-1), [me, ce] = z(
      o ?? (h ? [] : "")
    ), E = r ?? me, ie = X(null), xe = X(null), Fe = X(null), de = X(null), [ue, _e] = z({ top: 0, left: 0, width: 0 }), _ = Sn[f], ye = $n[m];
    V(() => {
      if (!K || !ie.current) return;
      const g = () => {
        const j = ie.current.getBoundingClientRect();
        _e({ top: j.bottom + 4, left: j.left, width: j.width });
      };
      return g(), window.addEventListener("scroll", g, !0), window.addEventListener("resize", g), () => {
        window.removeEventListener("scroll", g, !0), window.removeEventListener("resize", g);
      };
    }, [K]);
    const Re = P(
      (g) => {
        ne(g), w && (de.current && clearTimeout(de.current), de.current = setTimeout(() => {
          w(g);
        }, T));
      },
      [w, T]
    );
    V(() => () => {
      de.current && clearTimeout(de.current);
    }, []);
    const re = te(() => !O || w ? t : t.filter(
      (g) => {
        var j;
        return g.label.toLowerCase().includes(O.toLowerCase()) || ((j = g.description) == null ? void 0 : j.toLowerCase().includes(O.toLowerCase()));
      }
    ), [t, O, w]), G = te(() => {
      if (!D) return null;
      const g = {};
      return re.forEach((j) => {
        const A = j.group || "Other";
        g[A] || (g[A] = []), g[A].push(j);
      }), g;
    }, [re, D]), he = te(() => {
      var g;
      return h && Array.isArray(E) ? E.map((j) => {
        var A;
        return (A = t.find((ge) => ge.value === j)) == null ? void 0 : A.label;
      }).filter(Boolean) : (g = t.find((j) => j.value === E)) != null && g.label ? [t.find((j) => j.value === E).label] : [];
    }, [E, t, h]), C = P(
      (g) => h && Array.isArray(E) ? E.includes(g) : E === g,
      [E, h]
    ), B = P(
      (g) => {
        if (h) {
          const j = Array.isArray(E) ? [...E] : [], A = j.indexOf(g);
          if (A > -1)
            j.splice(A, 1);
          else {
            if (x && j.length >= x) return;
            j.push(g);
          }
          ce(j), n == null || n(j);
        } else
          ce(g), n == null || n(g), Q(!1);
        ne("");
      },
      [E, h, x, n]
    ), ee = te(
      () => re.filter((g) => !g.disabled).map((g) => g.value),
      [re]
    ), R = te(() => !h || !Array.isArray(E) ? !1 : ee.length > 0 && ee.every((g) => E.includes(g)), [h, E, ee]), We = te(() => !h || !Array.isArray(E) ? !1 : ee.some((g) => E.includes(g)) && !R, [h, E, ee, R]), Le = P(() => {
      if (h)
        if (R) {
          const g = Array.isArray(E) ? E.filter((j) => !ee.includes(j)) : [];
          ce(g), n == null || n(g);
        } else {
          const g = Array.isArray(E) ? [...E] : [];
          for (const j of ee)
            if (!g.includes(j)) {
              if (x && g.length >= x) break;
              g.push(j);
            }
          ce(g), n == null || n(g);
        }
    }, [h, R, E, ee, x, n]), He = P(() => {
      O.trim() && (W == null || W(O.trim()), ne(""));
    }, [O, W]), be = te(() => !I || !O.trim() ? !1 : !t.some((g) => g.label.toLowerCase() === O.toLowerCase()), [I, O, t]), Oe = P(
      (g) => {
        g.stopPropagation();
        const j = h ? [] : "";
        ce(j), n == null || n(j);
      },
      [h, n]
    ), je = D ? Object.values(G || {}).flat() : re, ot = P(
      (g) => {
        var j;
        if (!(p || u))
          switch (g.key) {
            case "ArrowDown":
              g.preventDefault(), K ? ae(
                (A) => A < je.length - 1 ? A + 1 : 0
              ) : Q(!0);
              break;
            case "ArrowUp":
              g.preventDefault(), ae(
                (A) => A > 0 ? A - 1 : je.length - 1
              );
              break;
            case "Enter":
            case " ":
              if (g.preventDefault(), K && oe >= 0) {
                const A = je[oe];
                A && !A.disabled && B(A.value);
              } else
                Q(!0);
              break;
            case "Escape":
              Q(!1), (j = ie.current) == null || j.focus();
              break;
          }
      },
      [p, u, K, oe, je, B]
    );
    V(() => {
      const g = (j) => {
        const A = j.target;
        ie.current && !ie.current.contains(A) && xe.current && !xe.current.contains(A) && (Q(!1), ne(""));
      };
      return document.addEventListener("mousedown", g), () => document.removeEventListener("mousedown", g);
    }, []), V(() => {
      K && S && setTimeout(() => {
        var g;
        return (g = Fe.current) == null ? void 0 : g.focus();
      }, 50), K && ae(-1);
    }, [K, S]);
    const Ue = h ? Array.isArray(E) && E.length > 0 : !!E, Ce = c ? "error" : v, Vt = () => {
      const g = he, j = M && M > 0 ? M : g.length, A = g.slice(0, j), ge = g.length - A.length;
      return /* @__PURE__ */ s("span", { className: "flex flex-wrap gap-1", children: [
        A.map((Ke) => /* @__PURE__ */ s(
          "span",
          {
            className: "inline-flex items-center gap-1 px-2 py-0.5 rounded-[var(--radius-sm)] bg-[var(--Colors--Background--bg-brand-secondary)] text-[var(--Colors--Text--text-brand-primary)]",
            style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
            children: [
              Ke,
              /* @__PURE__ */ e(
                J,
                {
                  size: 10,
                  className: "cursor-pointer hover:text-destructive transition-colors",
                  onClick: (er) => {
                    er.stopPropagation();
                    const lt = t.find((tr) => tr.label === Ke);
                    lt && B(lt.value);
                  }
                }
              )
            ]
          },
          Ke
        )),
        ge > 0 && /* @__PURE__ */ s(
          "span",
          {
            className: "inline-flex items-center px-2 py-0.5 rounded-[var(--radius-sm)] bg-muted text-muted-foreground",
            style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
            children: [
              "+",
              ge,
              " more"
            ]
          }
        )
      ] });
    }, at = (g, j) => {
      const A = C(g.value);
      return U ? /* @__PURE__ */ e(
        "button",
        {
          type: "button",
          role: "option",
          "aria-selected": A,
          disabled: g.disabled,
          className: `w-full ${_.option} transition-colors cursor-pointer
              ${A ? "bg-[var(--Colors--Background--bg-brand-secondary)]" : ""}
              ${oe === j && !A ? "bg-[var(--Colors--Background--bg-primary_hover)]" : ""}
              ${g.disabled ? "opacity-40 cursor-not-allowed" : "hover:bg-[var(--Colors--Background--bg-primary_hover)]"}
            `,
          style: _.optionStyle,
          onClick: () => !g.disabled && B(g.value),
          onMouseEnter: () => ae(j),
          children: U(g, A)
        },
        g.value
      ) : /* @__PURE__ */ s(
        "button",
        {
          type: "button",
          role: "option",
          "aria-selected": A,
          disabled: g.disabled,
          className: `w-full flex items-center gap-[var(--Spacing--Spacing-lg)] ${_.option} transition-colors cursor-pointer
            ${A ? "bg-[var(--Colors--Background--bg-brand-secondary)] text-[var(--Colors--Text--text-brand-primary)]" : ""}
            ${oe === j && !A ? "bg-[var(--Colors--Background--bg-primary_hover)]" : ""}
            ${g.disabled ? "opacity-40 cursor-not-allowed" : "hover:bg-[var(--Colors--Background--bg-primary_hover)]"}
          `,
          style: { ..._.optionStyle, ..._.textStyle },
          onClick: () => !g.disabled && B(g.value),
          onMouseEnter: () => ae(j),
          children: [
            h && /* @__PURE__ */ e(
              "span",
              {
                className: `flex-shrink-0 w-4 h-4 rounded-[var(--radius-sm)] border flex items-center justify-center transition-colors
                ${A ? "bg-primary border-primary" : "border-border"}
              `,
                children: A && /* @__PURE__ */ e(fe, { size: 10, className: "text-primary-foreground" })
              }
            ),
            g.avatar && /* @__PURE__ */ e(
              "img",
              {
                src: g.avatar,
                alt: "",
                className: "flex-shrink-0 rounded-full object-cover",
                style: { width: _.avatarSize, height: _.avatarSize }
              }
            ),
            g.icon && /* @__PURE__ */ e("span", { className: "flex-shrink-0", children: g.icon }),
            /* @__PURE__ */ s("span", { className: "flex-1 text-left", children: [
              /* @__PURE__ */ e("span", { className: "block", children: g.label }),
              g.description && /* @__PURE__ */ e(
                "span",
                {
                  className: "block text-muted-foreground",
                  style: { marginTop: "var(--Spacing--Spacing-xxs)", ..._.descStyle },
                  children: g.description
                }
              )
            ] }),
            !h && A && /* @__PURE__ */ e(fe, { size: _.icon, className: "text-primary flex-shrink-0" })
          ]
        },
        g.value
      );
    };
    let st = 0;
    return /* @__PURE__ */ s("div", { ref: Z, className: `relative w-full ${$}`, id: y, children: [
      i && /* @__PURE__ */ s("label", { className: "block mb-1.5 text-foreground", children: [
        i,
        b && /* @__PURE__ */ e("span", { className: "text-destructive ml-0.5", children: "*" })
      ] }),
      /* @__PURE__ */ s(
        "button",
        {
          ref: ie,
          type: "button",
          role: "combobox",
          "aria-expanded": K,
          "aria-haspopup": "listbox",
          disabled: p || u,
          onClick: () => !p && !u && Q(!K),
          onKeyDown: ot,
          className: `w-full flex items-center rounded-[var(--radius-md)] transition-all duration-150 ${_.trigger}
            ${ye.base}
            ${!p && !u ? ye.hover : ""}
            ${K ? ye.focus : ""}
            ${p ? "opacity-50 cursor-not-allowed" : "cursor-pointer"}
            ${u ? "opacity-70 cursor-wait" : ""}
            ${zn[Ce]}
          `,
          style: _.textStyle,
          children: [
            u ? /* @__PURE__ */ e(Se, { size: _.icon, className: "animate-spin text-muted-foreground" }) : null,
            /* @__PURE__ */ e("span", { className: `flex-1 text-left truncate ${Ue ? "text-foreground" : "text-muted-foreground"}`, children: Ue ? h ? Vt() : he[0] : a }),
            N && Ue && !p && /* @__PURE__ */ e(
              J,
              {
                size: _.icon,
                className: "flex-shrink-0 text-muted-foreground hover:text-foreground cursor-pointer transition-colors",
                onClick: Oe
              }
            ),
            /* @__PURE__ */ e(
              Ye,
              {
                size: _.icon,
                className: `flex-shrink-0 text-muted-foreground transition-transform duration-200 ${K ? "rotate-180" : ""}`
              }
            )
          ]
        }
      ),
      K && nt(
        /* @__PURE__ */ s(
          "div",
          {
            ref: xe,
            role: "listbox",
            className: "fixed z-[9999] bg-popover border border-border rounded-[var(--radius)] shadow-elevation-sm overflow-hidden",
            style: { top: ue.top, left: ue.left, width: ue.width || "auto" },
            children: [
              S && /* @__PURE__ */ e("div", { className: "p-2 border-b border-border", children: /* @__PURE__ */ s("div", { className: "relative", children: [
                /* @__PURE__ */ e(Be, { size: 14, className: "absolute left-2.5 top-1/2 -translate-y-1/2 text-muted-foreground" }),
                /* @__PURE__ */ e(
                  "input",
                  {
                    ref: Fe,
                    type: "text",
                    value: O,
                    onChange: (g) => Re(g.target.value),
                    placeholder: "Search\\u2026",
                    className: "w-full pl-8 pr-3 py-1.5 bg-background border border-border rounded-[var(--radius-sm)] outline-none placeholder:text-muted-foreground focus:border-ring focus:ring-1 focus:ring-ring/20 transition-colors",
                    style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
                    onKeyDown: ot
                  }
                )
              ] }) }),
              h && k && re.length > 0 && /* @__PURE__ */ s(
                "button",
                {
                  type: "button",
                  className: `w-full flex items-center gap-[var(--Spacing--Spacing-lg)] ${_.option} border-b border-border transition-colors hover:bg-[var(--Colors--Background--bg-primary_hover)] cursor-pointer`,
                  style: { ..._.optionStyle, ..._.textStyle },
                  onClick: Le,
                  children: [
                    /* @__PURE__ */ s(
                      "span",
                      {
                        className: `flex-shrink-0 w-4 h-4 rounded-[var(--radius-sm)] border flex items-center justify-center transition-colors
                      ${R ? "bg-primary border-primary" : We ? "bg-primary/50 border-primary" : "border-border"}
                    `,
                        children: [
                          R && /* @__PURE__ */ e(fe, { size: 10, className: "text-primary-foreground" }),
                          We && !R && /* @__PURE__ */ e("span", { className: "w-2 h-0.5 bg-primary-foreground rounded-full" })
                        ]
                      }
                    ),
                    /* @__PURE__ */ e("span", { className: "text-foreground", children: R ? "Deselect All" : "Select All" }),
                    /* @__PURE__ */ s(
                      "span",
                      {
                        className: "ml-auto text-muted-foreground",
                        style: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-label)" },
                        children: [
                          Array.isArray(E) ? E.length : 0,
                          "/",
                          re.filter((g) => !g.disabled).length
                        ]
                      }
                    )
                  ]
                }
              ),
              /* @__PURE__ */ s("div", { className: "max-h-60 overflow-y-auto p-1", children: [
                H ? /* @__PURE__ */ s("div", { className: "flex items-center justify-center py-6 gap-2 text-muted-foreground", children: [
                  /* @__PURE__ */ e(Se, { size: 16, className: "animate-spin" }),
                  /* @__PURE__ */ e("span", { style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)" }, children: "Searching\\u2026" })
                ] }) : re.length === 0 && !be ? Y ? /* @__PURE__ */ e("div", { className: "py-4 px-3", children: Y }) : /* @__PURE__ */ s("div", { className: "text-center py-6 text-muted-foreground flex flex-col items-center gap-2", children: [
                  /* @__PURE__ */ e(dr, { size: 24, className: "opacity-40" }),
                  /* @__PURE__ */ e("span", { style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)" }, children: "No options found" })
                ] }) : D && G ? Object.entries(G).map(([g, j]) => /* @__PURE__ */ s("div", { children: [
                  /* @__PURE__ */ e(
                    "div",
                    {
                      className: "px-3 py-1.5 uppercase tracking-wider text-muted-foreground",
                      style: { fontFamily: "var(--font-button)", fontSize: "var(--text-button)", fontWeight: "var(--weight-button)" },
                      children: g
                    }
                  ),
                  j.map((A) => {
                    const ge = at(A, st);
                    return st++, ge;
                  })
                ] }, g)) : re.map((g, j) => at(g, j)),
                be && /* @__PURE__ */ s(
                  "button",
                  {
                    type: "button",
                    className: `w-full flex items-center gap-[var(--Spacing--Spacing-lg)] ${_.option} text-primary transition-colors hover:bg-[var(--Colors--Background--bg-primary_hover)] cursor-pointer`,
                    style: { ..._.optionStyle, ..._.textStyle },
                    onClick: He,
                    children: [
                      /* @__PURE__ */ e(jt, { size: _.icon, className: "flex-shrink-0" }),
                      /* @__PURE__ */ e("span", { children: L.replace("{query}", O.trim()) })
                    ]
                  }
                )
              ] }),
              (F || h && x) && /* @__PURE__ */ s("div", { className: "border-t border-border", children: [
                h && x && /* @__PURE__ */ s(
                  "div",
                  {
                    className: "px-3 py-2 text-muted-foreground",
                    style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
                    children: [
                      Array.isArray(E) ? E.length : 0,
                      " / ",
                      x,
                      " selected"
                    ]
                  }
                ),
                F && /* @__PURE__ */ e("div", { className: "px-3 py-2", children: F })
              ] })
            ]
          }
        ),
        document.body
      ),
      (l || c || d) && /* @__PURE__ */ s("div", { className: "mt-1.5 flex items-center gap-1", children: [
        Ce === "error" && /* @__PURE__ */ e(ke, { size: 12, className: "text-destructive flex-shrink-0" }),
        /* @__PURE__ */ e(
          "span",
          {
            className: `${Ce === "error" ? "text-destructive" : Ce === "success" ? "text-chart-2" : "text-muted-foreground"}`,
            style: { fontFamily: "var(--font-label)", fontSize: "var(--text-label)", fontWeight: "var(--weight-label)" },
            children: c || d || l
          }
        )
      ] })
    ] });
  }
);
kn.displayName = "Dropdown";
const Fn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, gt = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
};
function Vo({
  icon: t,
  title: r,
  description: o,
  action: n,
  secondaryAction: a,
  size: i = "md"
}) {
  return /* @__PURE__ */ s("div", { className: `flex flex-col items-center text-center ${i === "sm" ? "py-6 px-4" : i === "lg" ? "py-14 px-8" : "py-10 px-6"}`, children: [
    t && /* @__PURE__ */ e(
      "div",
      {
        className: `${i === "sm" ? "w-10 h-10" : i === "lg" ? "w-16 h-16" : "w-12 h-12"} rounded-full bg-muted flex items-center justify-center mb-4`,
        children: /* @__PURE__ */ e("span", { className: "text-muted-foreground", children: t })
      }
    ),
    /* @__PURE__ */ e("h4", { className: "text-foreground", children: r }),
    o && /* @__PURE__ */ e(
      "p",
      {
        className: "text-muted-foreground mt-1.5 max-w-sm",
        style: Fn,
        children: o
      }
    ),
    (n || a) && /* @__PURE__ */ s("div", { className: "flex items-center gap-3 mt-5", children: [
      n && /* @__PURE__ */ s(
        "button",
        {
          onClick: n.onClick,
          className: "inline-flex items-center gap-2 px-4 py-2 rounded-[var(--radius-md)] bg-primary text-primary-foreground hover:bg-primary/90 transition-colors cursor-pointer",
          style: gt,
          children: [
            n.icon,
            n.label
          ]
        }
      ),
      a && /* @__PURE__ */ e(
        "button",
        {
          onClick: a.onClick,
          className: "inline-flex items-center gap-2 px-4 py-2 rounded-[var(--radius-md)] border border-border text-foreground hover:bg-accent transition-colors cursor-pointer",
          style: gt,
          children: a.label
        }
      )
    ] })
  ] });
}
const Wn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, De = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, Ot = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, Ln = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
};
function Ut(t) {
  return t < 1024 ? t + " B" : t < 1024 * 1024 ? (t / 1024).toFixed(1) + " KB" : (t / (1024 * 1024)).toFixed(1) + " MB";
}
function jn(t) {
  return t.startsWith("image/") ? /* @__PURE__ */ e(ur, { size: 16, className: "text-chart-2" }) : t.startsWith("video/") ? /* @__PURE__ */ e(fr, { size: 16, className: "text-chart-5" }) : t.includes("pdf") || t.includes("document") ? /* @__PURE__ */ e(mr, { size: 16, className: "text-destructive" }) : /* @__PURE__ */ e(Ct, { size: 16, className: "text-muted-foreground" });
}
function pt({
  files: t,
  onRemove: r
}) {
  return /* @__PURE__ */ e("div", { className: "space-y-2", children: t.map((o) => /* @__PURE__ */ s(
    "div",
    {
      className: "flex items-center gap-3 px-3 py-2 rounded-[var(--radius-md)] border border-border bg-card",
      children: [
        jn(o.type),
        /* @__PURE__ */ s("div", { className: "flex-1 min-w-0", children: [
          /* @__PURE__ */ e("p", { className: "text-foreground truncate", style: Wn, children: o.name }),
          /* @__PURE__ */ s("div", { className: "flex items-center gap-2", children: [
            /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: Ot, children: Ut(o.size) }),
            o.status === "uploading" && /* @__PURE__ */ e("div", { className: "flex-1 h-1 bg-muted rounded-full overflow-hidden", children: /* @__PURE__ */ e(
              "div",
              {
                className: "h-full bg-primary rounded-full transition-all",
                style: { width: `${o.progress}%` }
              }
            ) }),
            o.status === "done" && /* @__PURE__ */ e(se, { size: 12, className: "text-chart-2" })
          ] })
        ] }),
        /* @__PURE__ */ e(
          "button",
          {
            onClick: () => r(o.id),
            className: "text-muted-foreground hover:text-destructive cursor-pointer",
            children: /* @__PURE__ */ e(J, { size: 14 })
          }
        )
      ]
    },
    o.id
  )) });
}
function ea({
  accept: t,
  maxSize: r,
  multiple: o,
  disabled: n,
  variant: a = "dropzone",
  label: i,
  description: l,
  value: c,
  onChange: d,
  onUpload: f,
  onRemove: m
}) {
  const [v, p] = z([]), [u, S] = z(!1), N = X(null), h = c ?? v, x = (b) => {
    if (!b) return;
    const y = Array.from(b);
    if (d && d(y), c === void 0) {
      const $ = y.map((k) => ({
        id: Math.random().toString(36).slice(2),
        name: k.name,
        size: k.size,
        type: k.type,
        progress: 0,
        status: "uploading"
      }));
      p((k) => [...k, ...$]), $.forEach((k) => {
        let M = 0;
        const I = setInterval(() => {
          M += Math.random() * 30 + 10, M >= 100 && (M = 100, clearInterval(I)), p(
            (W) => W.map(
              (L) => L.id === k.id ? {
                ...L,
                progress: Math.min(M, 100),
                status: M >= 100 ? "done" : "uploading"
              } : L
            )
          ), M >= 100 && f && f({ ...k, progress: 100, status: "done" });
        }, 300);
      });
    }
  }, D = (b) => {
    m && m(b), c === void 0 && p((y) => y.filter(($) => $.id !== b));
  };
  return a === "button" ? /* @__PURE__ */ s("div", { className: "space-y-3", children: [
    i && /* @__PURE__ */ e("label", { className: "block text-foreground", style: De, children: i }),
    /* @__PURE__ */ s(
      "button",
      {
        onClick: () => {
          var b;
          return (b = N.current) == null ? void 0 : b.click();
        },
        disabled: n,
        className: `inline-flex items-center gap-2 px-4 py-2 rounded-[var(--radius-md)] border border-border bg-background text-foreground hover:bg-accent transition-colors cursor-pointer ${n ? "opacity-50 pointer-events-none" : ""}`,
        style: Ln,
        children: [
          /* @__PURE__ */ e(Xe, { size: 14 }),
          " Choose File",
          o ? "s" : ""
        ]
      }
    ),
    /* @__PURE__ */ e(
      "input",
      {
        ref: N,
        type: "file",
        className: "hidden",
        accept: t,
        multiple: o,
        onChange: (b) => x(b.target.files)
      }
    ),
    h.length > 0 && /* @__PURE__ */ e(pt, { files: h, onRemove: D })
  ] }) : a === "avatar" ? /* @__PURE__ */ s("div", { className: "space-y-2", children: [
    i && /* @__PURE__ */ e("label", { className: "block text-foreground", style: De, children: i }),
    /* @__PURE__ */ e(
      "div",
      {
        onClick: () => {
          var b;
          return !n && ((b = N.current) == null ? void 0 : b.click());
        },
        className: `w-24 h-24 rounded-full border-2 border-dashed flex items-center justify-center cursor-pointer transition-colors
            ${n ? "opacity-50 pointer-events-none" : ""} ${u ? "border-primary bg-primary/5" : "border-border hover:border-primary/40"}`,
        children: h.length > 0 && h[h.length - 1].status === "done" ? /* @__PURE__ */ e(se, { size: 24, className: "text-chart-2" }) : /* @__PURE__ */ e(Xe, { size: 20, className: "text-muted-foreground" })
      }
    ),
    /* @__PURE__ */ e(
      "input",
      {
        ref: N,
        type: "file",
        className: "hidden",
        accept: t || "image/*",
        onChange: (b) => x(b.target.files)
      }
    )
  ] }) : /* @__PURE__ */ s("div", { className: "space-y-3", children: [
    i && /* @__PURE__ */ e("label", { className: "block text-foreground", style: De, children: i }),
    /* @__PURE__ */ s(
      "div",
      {
        onDragOver: (b) => {
          b.preventDefault(), S(!0);
        },
        onDragLeave: () => S(!1),
        onDrop: (b) => {
          b.preventDefault(), S(!1), x(b.dataTransfer.files);
        },
        onClick: () => {
          var b;
          return !n && ((b = N.current) == null ? void 0 : b.click());
        },
        className: `relative flex flex-col items-center justify-center gap-2 px-6 py-10 rounded-[var(--radius-lg)] border-2 border-dashed transition-all cursor-pointer
          ${n ? "opacity-50 pointer-events-none" : ""}
          ${u ? "border-primary bg-primary/5" : "border-border hover:border-primary/40 hover:bg-accent/30"}`,
        children: [
          /* @__PURE__ */ e("div", { className: "w-12 h-12 rounded-full bg-primary/10 flex items-center justify-center", children: /* @__PURE__ */ e(Xe, { size: 20, className: "text-primary" }) }),
          /* @__PURE__ */ s("div", { className: "text-center", children: [
            /* @__PURE__ */ s("p", { className: "text-foreground", style: De, children: [
              /* @__PURE__ */ e("span", { className: "text-primary", children: "Click to upload" }),
              " or drag and drop"
            ] }),
            /* @__PURE__ */ s("p", { className: "text-muted-foreground", style: Ot, children: [
              l || (t ? `Accepted: ${t}` : "Any file type"),
              r && ` · Max ${Ut(r)}`
            ] })
          ] })
        ]
      }
    ),
    /* @__PURE__ */ e(
      "input",
      {
        ref: N,
        type: "file",
        className: "hidden",
        accept: t,
        multiple: o,
        onChange: (b) => x(b.target.files)
      }
    ),
    h.length > 0 && /* @__PURE__ */ e(pt, { files: h, onRemove: D })
  ] });
}
const Cn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, Mn = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
};
function ta({ images: t, initialIndex: r = 0 }) {
  const [o, n] = z(!1), [a, i] = z(r), [l, c] = z(1), [d, f] = z(0), m = t[a], v = () => {
    i((h) => (h - 1 + t.length) % t.length), c(1), f(0);
  }, p = () => {
    i((h) => (h + 1) % t.length), c(1), f(0);
  }, u = () => c((h) => Math.min(h + 0.25, 3)), S = () => c((h) => Math.max(h - 0.25, 0.5)), N = () => f((h) => (h + 90) % 360);
  return /* @__PURE__ */ s("div", { children: [
    /* @__PURE__ */ e("div", { className: "flex flex-wrap gap-2", children: t.map((h, x) => /* @__PURE__ */ e(
      "button",
      {
        onClick: () => {
          i(x), n(!0), c(1), f(0);
        },
        className: `rounded-[var(--radius-md)] overflow-hidden border-2 transition-all cursor-pointer hover:opacity-80
              ${a === x && o ? "border-primary" : "border-border"}`,
        children: /* @__PURE__ */ e("div", { className: "w-20 h-20 bg-muted flex items-center justify-center overflow-hidden", children: /* @__PURE__ */ e(
          "img",
          {
            src: h.thumbnail || h.src,
            alt: h.alt,
            className: "w-full h-full object-cover",
            loading: "lazy"
          }
        ) })
      },
      x
    )) }),
    o && /* @__PURE__ */ s("div", { className: "fixed inset-0 z-[60] bg-foreground/80 flex flex-col", children: [
      /* @__PURE__ */ s("div", { className: "flex items-center justify-between px-4 py-3 bg-foreground/20", children: [
        /* @__PURE__ */ s("span", { className: "text-white", style: Cn, children: [
          m.alt,
          " (",
          a + 1,
          "/",
          t.length,
          ")"
        ] }),
        /* @__PURE__ */ s("div", { className: "flex items-center gap-1", children: [
          /* @__PURE__ */ e(
            "button",
            {
              onClick: S,
              className: "w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] text-white/70 hover:text-white hover:bg-white/10 cursor-pointer",
              children: /* @__PURE__ */ e(hr, { size: 16 })
            }
          ),
          /* @__PURE__ */ s("span", { className: "text-white/70 px-2", style: Mn, children: [
            Math.round(l * 100),
            "%"
          ] }),
          /* @__PURE__ */ e(
            "button",
            {
              onClick: u,
              className: "w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] text-white/70 hover:text-white hover:bg-white/10 cursor-pointer",
              children: /* @__PURE__ */ e(br, { size: 16 })
            }
          ),
          /* @__PURE__ */ e(
            "button",
            {
              onClick: N,
              className: "w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] text-white/70 hover:text-white hover:bg-white/10 cursor-pointer",
              children: /* @__PURE__ */ e(gr, { size: 16 })
            }
          ),
          /* @__PURE__ */ e("div", { className: "w-px h-5 bg-white/20 mx-1" }),
          /* @__PURE__ */ e(
            "button",
            {
              onClick: () => {
                n(!1), c(1), f(0);
              },
              className: "w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] text-white/70 hover:text-white hover:bg-white/10 cursor-pointer",
              children: /* @__PURE__ */ e(J, { size: 16 })
            }
          )
        ] })
      ] }),
      /* @__PURE__ */ s(
        "div",
        {
          className: "flex-1 flex items-center justify-center relative overflow-hidden",
          onClick: (h) => {
            h.target === h.currentTarget && (n(!1), c(1), f(0));
          },
          children: [
            t.length > 1 && /* @__PURE__ */ e(
              "button",
              {
                onClick: v,
                className: "absolute left-4 w-10 h-10 rounded-full bg-foreground/30 text-white flex items-center justify-center hover:bg-foreground/50 cursor-pointer z-10",
                children: /* @__PURE__ */ e($e, { size: 20 })
              }
            ),
            /* @__PURE__ */ e(
              "img",
              {
                src: m.src,
                alt: m.alt,
                className: "max-w-[90%] max-h-[80vh] object-contain transition-transform",
                style: {
                  transform: `scale(${l}) rotate(${d}deg)`
                }
              }
            ),
            t.length > 1 && /* @__PURE__ */ e(
              "button",
              {
                onClick: p,
                className: "absolute right-4 w-10 h-10 rounded-full bg-foreground/30 text-white flex items-center justify-center hover:bg-foreground/50 cursor-pointer z-10",
                children: /* @__PURE__ */ e(le, { size: 20 })
              }
            )
          ]
        }
      ),
      t.length > 1 && /* @__PURE__ */ e("div", { className: "flex items-center justify-center gap-2 py-3 bg-foreground/20", children: t.map((h, x) => /* @__PURE__ */ e(
        "button",
        {
          onClick: () => {
            i(x), c(1), f(0);
          },
          className: `w-12 h-12 rounded-[var(--radius-sm)] overflow-hidden border-2 cursor-pointer transition-all
                    ${x === a ? "border-white" : "border-transparent opacity-50 hover:opacity-80"}`,
          children: /* @__PURE__ */ e(
            "img",
            {
              src: h.thumbnail || h.src,
              alt: h.alt,
              className: "w-full h-full object-cover"
            }
          )
        },
        x
      )) })
    ] })
  ] });
}
const Dn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, An = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, In = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
};
function Tn({ items: t, parentRect: r }) {
  const o = X(null), [n, a] = z({ top: r.top, left: r.right + 4 });
  return V(() => {
    if (!o.current) return;
    const l = o.current.getBoundingClientRect();
    let c = r.top, d = r.right + 4;
    d + l.width > window.innerWidth - 8 && (d = r.left - l.width - 4), c + l.height > window.innerHeight - 8 && (c = window.innerHeight - l.height - 8), a({ top: c, left: d });
  }, [r]), nt(
    /* @__PURE__ */ e(
      "div",
      {
        ref: o,
        className: "fixed z-[9999] min-w-[200px] rounded-[var(--radius)] border border-border bg-popover shadow-md py-1",
        style: { top: n.top, left: n.left },
        children: t.map((i, l) => /* @__PURE__ */ e(Kt, { item: i }, l))
      }
    ),
    document.body
  );
}
function Kt({ item: t }) {
  const [r, o] = z(!1), n = X(null), a = X(null);
  if (t.divider) return /* @__PURE__ */ e("div", { className: "my-1 border-t border-border" });
  if (t.label)
    return /* @__PURE__ */ e("div", { className: "px-3 py-1.5 text-muted-foreground uppercase tracking-wider", style: In, children: t.label });
  const i = t.children && t.children.length > 0;
  return /* @__PURE__ */ s("div", { ref: n, className: "relative", onMouseEnter: () => {
    a.current && clearTimeout(a.current), i && o(!0);
  }, onMouseLeave: () => {
    a.current = setTimeout(() => o(!1), 150);
  }, children: [
    /* @__PURE__ */ s(
      "button",
      {
        onClick: () => {
          var d;
          i || (d = t.onClick) == null || d.call(t);
        },
        disabled: t.disabled,
        className: `w-full flex items-center text-left rounded-[var(--radius-sm)] transition-colors cursor-pointer ${t.destructive ? "text-destructive hover:bg-destructive/10" : t.disabled ? "text-muted-foreground opacity-50 cursor-not-allowed" : "text-popover-foreground hover:bg-[var(--Colors--Background--bg-primary_hover)]"}`,
        style: { padding: "var(--Spacing--Spacing-md) var(--Spacing--Spacing-2xl)", gap: "var(--Spacing--Spacing-lg)", ...Dn },
        children: [
          t.icon && /* @__PURE__ */ e("span", { className: "flex-shrink-0", children: t.icon }),
          /* @__PURE__ */ e("span", { className: "flex-1", children: t.text }),
          t.shortcut && /* @__PURE__ */ e("span", { className: "text-muted-foreground ml-4", style: An, children: t.shortcut }),
          i && /* @__PURE__ */ e(le, { size: 14, className: "text-muted-foreground" })
        ]
      }
    ),
    i && r && n.current && /* @__PURE__ */ e(Tn, { items: t.children, parentRect: n.current.getBoundingClientRect() })
  ] });
}
function En({ items: t, open: r, onClose: o, triggerRef: n, className: a = "" }) {
  const i = X(null), [l, c] = z({ top: 0, left: 0 });
  return V(() => {
    if (!r || !n.current) return;
    const d = n.current.getBoundingClientRect();
    let f = d.bottom + 4, m = d.left;
    requestAnimationFrame(() => {
      if (!i.current) return;
      const v = i.current.getBoundingClientRect();
      m + v.width > window.innerWidth - 8 && (m = d.right - v.width), f + v.height > window.innerHeight - 8 && (f = d.top - v.height - 4), c({ top: f, left: m });
    }), c({ top: f, left: m });
  }, [r, n]), V(() => {
    if (!r) return;
    const d = (f) => {
      var m, v;
      (m = n.current) != null && m.contains(f.target) || (v = i.current) != null && v.contains(f.target) || o();
    };
    return document.addEventListener("mousedown", d), () => document.removeEventListener("mousedown", d);
  }, [r, o, n]), r ? nt(
    /* @__PURE__ */ s(
      "div",
      {
        ref: i,
        className: `fixed z-[9999] min-w-[200px] rounded-[var(--radius)] border border-border bg-popover shadow-md py-1 animate-[fadeIn_0.15s_ease] ${a}`,
        style: { top: l.top, left: l.left },
        children: [
          t.map((d, f) => /* @__PURE__ */ e(Kt, { item: d }, f)),
          /* @__PURE__ */ e("style", { children: "@keyframes fadeIn { from { opacity: 0; transform: translateY(-4px); } to { opacity: 1; transform: translateY(0); } }" })
        ]
      }
    ),
    document.body
  ) : null;
}
En.displayName = "Menu";
const Xt = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, Pe = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, Bn = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, Te = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
};
function Yn({
  type: t = "info",
  title: r,
  message: o,
  closable: n = !0,
  onClose: a,
  action: i,
  avatar: l,
  time: c,
  read: d
}) {
  const f = {
    info: /* @__PURE__ */ e(Wt, { size: 16, className: "text-primary" }),
    success: /* @__PURE__ */ e(se, { size: 16, className: "text-chart-2" }),
    warning: /* @__PURE__ */ e(Ft, { size: 16, className: "text-chart-5" }),
    error: /* @__PURE__ */ e(kt, { size: 16, className: "text-destructive" })
  };
  return /* @__PURE__ */ s(
    "div",
    {
      className: `flex gap-3 px-4 py-3 rounded-[var(--radius-md)] border border-border bg-card border-l-[3px] ${{
        info: "border-l-primary",
        success: "border-l-chart-2",
        warning: "border-l-chart-5",
        error: "border-l-destructive"
      }[t]} ${d === !1 ? "bg-primary/5" : ""}`,
      children: [
        l ? /* @__PURE__ */ e(
          "div",
          {
            className: "w-8 h-8 rounded-full bg-muted flex-shrink-0 flex items-center justify-center overflow-hidden",
            style: Pe,
            children: /* @__PURE__ */ e("span", { className: "text-primary", children: l.slice(0, 2).toUpperCase() })
          }
        ) : /* @__PURE__ */ e("span", { className: "flex-shrink-0 mt-0.5", children: f[t] }),
        /* @__PURE__ */ s("div", { className: "flex-1 min-w-0", children: [
          /* @__PURE__ */ s("div", { className: "flex items-start justify-between gap-2", children: [
            /* @__PURE__ */ e("span", { className: "text-foreground", style: Pe, children: r }),
            /* @__PURE__ */ s("div", { className: "flex items-center gap-1 flex-shrink-0", children: [
              c && /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: Bn, children: c }),
              d === !1 && /* @__PURE__ */ e("span", { className: "w-2 h-2 rounded-full bg-primary" }),
              n && a && /* @__PURE__ */ e(
                "button",
                {
                  onClick: a,
                  className: "text-muted-foreground hover:text-foreground cursor-pointer",
                  children: /* @__PURE__ */ e(J, { size: 14 })
                }
              )
            ] })
          ] }),
          o && /* @__PURE__ */ e("p", { className: "text-muted-foreground mt-0.5", style: Xt, children: o }),
          i && /* @__PURE__ */ s(
            "button",
            {
              onClick: i.onClick,
              className: "inline-flex items-center gap-1 text-primary mt-2 hover:underline cursor-pointer",
              style: Te,
              children: [
                i.label,
                " ",
                /* @__PURE__ */ e(pr, { size: 12 })
              ]
            }
          )
        ] })
      ]
    }
  );
}
function ra({
  items: t,
  onMarkRead: r,
  onMarkAllRead: o,
  onDismiss: n,
  onClearAll: a
}) {
  const [i, l] = z(t), c = r !== void 0 || o !== void 0, d = c ? t : i, f = d.filter((u) => !u.read).length, m = () => {
    o && o(), c || l((u) => u.map((S) => ({ ...S, read: !0 })));
  }, v = (u) => {
    n && n(u), c || l((S) => S.filter((N) => N.id !== u));
  }, p = (u) => {
    r && r(u), c || l(
      (S) => S.map((N) => N.id === u ? { ...N, read: !0 } : N)
    );
  };
  return /* @__PURE__ */ s("div", { className: "max-w-md rounded-[var(--radius-lg)] border border-border bg-card overflow-hidden", children: [
    /* @__PURE__ */ s("div", { className: "px-4 py-3 border-b border-border flex items-center justify-between", children: [
      /* @__PURE__ */ s("div", { className: "flex items-center gap-2", children: [
        /* @__PURE__ */ e(Mt, { size: 16, className: "text-foreground" }),
        /* @__PURE__ */ e("span", { className: "text-foreground", style: Pe, children: "Notifications" }),
        f > 0 && /* @__PURE__ */ e(
          "span",
          {
            className: "px-1.5 py-0.5 rounded-full bg-primary text-primary-foreground",
            style: Te,
            children: f
          }
        )
      ] }),
      /* @__PURE__ */ s("div", { className: "flex items-center gap-2", children: [
        f > 0 && /* @__PURE__ */ e(
          "button",
          {
            onClick: m,
            className: "text-primary cursor-pointer",
            style: Te,
            children: "Mark all read"
          }
        ),
        d.length > 0 && a && /* @__PURE__ */ e(
          "button",
          {
            onClick: a,
            className: "text-muted-foreground hover:text-destructive cursor-pointer",
            style: Te,
            children: "Clear all"
          }
        )
      ] })
    ] }),
    /* @__PURE__ */ e("div", { className: "divide-y divide-border max-h-[360px] overflow-y-auto", children: d.map((u) => /* @__PURE__ */ e(
      "div",
      {
        className: `px-3 py-2 ${u.read ? "" : "bg-primary/5"}`,
        onClick: () => !u.read && p(u.id),
        children: /* @__PURE__ */ e(
          Yn,
          {
            type: u.type,
            title: u.title,
            message: u.message,
            time: u.time,
            read: u.read,
            closable: !!n,
            onClose: () => v(u.id)
          }
        )
      },
      u.id
    )) }),
    d.length === 0 && /* @__PURE__ */ s("div", { className: "py-10 text-center text-muted-foreground", children: [
      /* @__PURE__ */ e(vr, { size: 24, className: "mx-auto mb-2 opacity-40" }),
      /* @__PURE__ */ e("span", { style: Xt, children: "No notifications" })
    ] })
  ] });
}
const _n = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
};
function na({
  trigger: t,
  children: r,
  placement: o = "bottom",
  title: n,
  open: a,
  onOpenChange: i
}) {
  const [l, c] = z(!1), d = a !== void 0 ? a : l, f = (p) => {
    i == null || i(p), a === void 0 && c(p);
  }, m = X(null);
  return V(() => {
    if (!d) return;
    const p = (u) => {
      m.current && !m.current.contains(u.target) && f(!1);
    };
    return document.addEventListener("mousedown", p), () => document.removeEventListener("mousedown", p);
  }, [d]), /* @__PURE__ */ s("div", { className: "relative inline-block", ref: m, children: [
    /* @__PURE__ */ e("div", { onClick: () => f(!d), className: "cursor-pointer", children: t }),
    d && /* @__PURE__ */ e(
      "div",
      {
        className: `absolute z-50 ${{
          top: "bottom-full left-1/2 -translate-x-1/2 mb-2",
          bottom: "top-full left-1/2 -translate-x-1/2 mt-2",
          left: "right-full top-1/2 -translate-y-1/2 mr-2",
          right: "left-full top-1/2 -translate-y-1/2 ml-2"
        }[o]} w-max max-w-xs`,
        children: /* @__PURE__ */ s("div", { className: "rounded-[var(--radius-lg)] border border-border bg-card shadow-lg overflow-hidden", children: [
          n && /* @__PURE__ */ s("div", { className: "px-4 py-2.5 border-b border-border flex items-center justify-between", children: [
            /* @__PURE__ */ e("span", { className: "text-foreground", style: _n, children: n }),
            /* @__PURE__ */ e(
              "button",
              {
                onClick: () => f(!1),
                className: "text-muted-foreground hover:text-foreground cursor-pointer",
                children: /* @__PURE__ */ e(J, { size: 14 })
              }
            )
          ] }),
          /* @__PURE__ */ e("div", { className: "px-4 py-3", children: r })
        ] })
      }
    )
  ] });
}
const Rn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)"
}, Hn = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
};
function On({
  value: t,
  max: r = 100,
  size: o = "md",
  color: n,
  label: a,
  showValue: i,
  indeterminate: l,
  className: c = ""
}) {
  const d = { sm: "h-1.5", md: "h-2.5", lg: "h-4" }, f = l ? 0 : Math.min(100, Math.max(0, (t ?? 0) / r * 100));
  return /* @__PURE__ */ s("div", { className: `w-full ${c}`, children: [
    (a || i) && /* @__PURE__ */ s("div", { className: "flex items-center justify-between mb-1.5", children: [
      a && /* @__PURE__ */ e("span", { className: "text-foreground", style: Rn, children: a }),
      i && !l && /* @__PURE__ */ s("span", { className: "text-muted-foreground", style: Hn, children: [
        Math.round(f),
        "%"
      ] })
    ] }),
    /* @__PURE__ */ e("div", { className: `w-full ${d[o]} rounded-full bg-muted overflow-hidden`, children: l ? /* @__PURE__ */ e(
      "div",
      {
        className: "h-full w-1/3 rounded-full animate-[indeterminate_1.5s_infinite_ease-in-out]",
        style: { backgroundColor: n ?? "var(--primary)" }
      }
    ) : /* @__PURE__ */ e(
      "div",
      {
        className: "h-full rounded-full transition-all duration-500",
        style: {
          width: `${f}%`,
          backgroundColor: n ?? "var(--primary)"
        }
      }
    ) }),
    /* @__PURE__ */ e("style", { children: "@keyframes indeterminate { 0% { transform: translateX(-100%); } 100% { transform: translateX(400%); } }" })
  ] });
}
On.displayName = "ProgressBar";
const Un = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, Kn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
};
function oa({
  value: t = 0,
  onChange: r,
  max: o = 5,
  size: n = "md",
  disabled: a,
  readOnly: i,
  icon: l = "star",
  showValue: c,
  label: d
}) {
  const [f, m] = z(0), v = n === "sm" ? 16 : n === "lg" ? 28 : 20, p = n === "sm" ? "gap-0.5" : n === "lg" ? "gap-1.5" : "gap-1", u = l === "heart" ? xr : l === "thumb" ? yr : wr, S = l === "heart" ? "text-destructive" : "text-chart-5", N = Array.from({ length: o }, (h, x) => x + 1);
  return /* @__PURE__ */ s(
    "div",
    {
      className: `inline-flex items-center ${p} ${a ? "opacity-50 pointer-events-none" : ""}`,
      children: [
        d && /* @__PURE__ */ e("span", { className: "text-foreground mr-2", style: Un, children: d }),
        N.map((h) => {
          const x = h <= (f || t);
          return /* @__PURE__ */ e(
            "button",
            {
              type: "button",
              className: `transition-transform ${i || a ? "" : "cursor-pointer hover:scale-110"} ${x ? S : "text-muted-foreground/40"}`,
              onMouseEnter: () => !i && !a && m(h),
              onMouseLeave: () => m(0),
              onClick: () => !i && !a && (r == null ? void 0 : r(h === t ? 0 : h)),
              disabled: a || i,
              children: /* @__PURE__ */ e(
                u,
                {
                  size: v,
                  fill: x ? "currentColor" : "none"
                }
              )
            },
            h
          );
        }),
        c && /* @__PURE__ */ s("span", { className: "text-foreground ml-1", style: Kn, children: [
          t,
          "/",
          o
        ] })
      ]
    }
  );
}
const Xn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, Gn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)"
}, qe = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
}, qn = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
};
function Jn({
  brand: t,
  groups: r,
  activeItem: o,
  onNavigate: n,
  collapsed: a = !1,
  onCollapsedChange: i,
  className: l = ""
}) {
  return /* @__PURE__ */ s(
    "div",
    {
      className: `bg-sidebar border-r border-sidebar-border flex flex-col flex-shrink-0 transition-all ${a ? "w-16" : "w-64"} ${l}`,
      children: [
        t && /* @__PURE__ */ s("div", { className: "px-4 py-4 border-b border-sidebar-border flex items-center gap-2", children: [
          t.logo ? /* @__PURE__ */ e(
            "img",
            {
              src: t.logo,
              alt: t.name,
              className: "w-8 h-8 rounded-[var(--radius-sm)] flex-shrink-0 object-cover"
            }
          ) : /* @__PURE__ */ e(
            "div",
            {
              className: "w-8 h-8 rounded-[var(--radius-sm)] bg-primary flex items-center justify-center text-primary-foreground flex-shrink-0",
              style: qe,
              children: t.name.charAt(0).toUpperCase()
            }
          ),
          !a && /* @__PURE__ */ e("span", { className: "text-sidebar-foreground truncate", style: Xn, children: t.name })
        ] }),
        /* @__PURE__ */ e("nav", { className: "flex-1 overflow-y-auto py-3 px-2", children: r.map((c) => /* @__PURE__ */ s("div", { className: "mb-4", children: [
          !a && /* @__PURE__ */ e(
            "span",
            {
              className: "px-2 mb-1.5 block text-muted-foreground uppercase tracking-wider",
              style: qe,
              children: c.label
            }
          ),
          /* @__PURE__ */ e("div", { className: "space-y-0.5", children: c.items.map((d) => {
            const f = o === d.id;
            return /* @__PURE__ */ s(
              "button",
              {
                onClick: () => n == null ? void 0 : n(d),
                className: `w-full flex items-center gap-2 px-2 py-2 rounded-[var(--radius-md)] transition-colors cursor-pointer ${f ? "bg-sidebar-accent text-sidebar-accent-foreground" : "text-sidebar-foreground hover:bg-sidebar-accent/50"}`,
                style: Gn,
                children: [
                  /* @__PURE__ */ e("span", { className: f ? "text-sidebar-primary" : "text-muted-foreground", children: d.icon }),
                  !a && /* @__PURE__ */ e("span", { className: "truncate flex-1 text-left", children: d.label }),
                  !a && d.badge && /* @__PURE__ */ e(
                    "span",
                    {
                      className: "px-1.5 py-0.5 rounded-full bg-primary text-primary-foreground",
                      style: { ...qe, lineHeight: "1" },
                      children: d.badge
                    }
                  )
                ]
              },
              d.id
            );
          }) })
        ] }, c.label)) }),
        i && /* @__PURE__ */ e("div", { className: "px-3 py-3 border-t border-sidebar-border", children: /* @__PURE__ */ e(
          "button",
          {
            onClick: () => i(!a),
            className: "w-full px-2 py-1.5 rounded-[var(--radius-md)] text-muted-foreground hover:text-foreground hover:bg-sidebar-accent/50 cursor-pointer text-left",
            style: qn,
            children: a ? ">" : "Collapse"
          }
        ) })
      ]
    }
  );
}
Jn.displayName = "Sidebar";
function q({
  width: t,
  height: r,
  variant: o = "rectangular",
  animate: n = !0,
  className: a
}) {
  return /* @__PURE__ */ e(
    "div",
    {
      className: `bg-muted ${n ? "animate-pulse" : ""} ${o === "circular" ? "rounded-full" : o === "rounded" ? "rounded-[var(--radius-md)]" : "rounded-[var(--radius-sm)]"} ${a ?? ""}`,
      style: {
        width: t ?? "100%",
        height: r ?? (o === "text" ? "1em" : o === "circular" ? 40 : 20)
      }
    }
  );
}
function aa() {
  return /* @__PURE__ */ s("div", { className: "rounded-[var(--radius-lg)] border border-border bg-card p-4 space-y-4 w-full max-w-sm", children: [
    /* @__PURE__ */ e(q, { variant: "rounded", height: 160 }),
    /* @__PURE__ */ s("div", { className: "space-y-2", children: [
      /* @__PURE__ */ e(q, { variant: "text", width: "60%", height: 20 }),
      /* @__PURE__ */ e(q, { variant: "text", width: "100%", height: 14 }),
      /* @__PURE__ */ e(q, { variant: "text", width: "80%", height: 14 })
    ] }),
    /* @__PURE__ */ s("div", { className: "flex items-center gap-3", children: [
      /* @__PURE__ */ e(q, { variant: "circular", width: 36, height: 36 }),
      /* @__PURE__ */ s("div", { className: "flex-1 space-y-1.5", children: [
        /* @__PURE__ */ e(q, { variant: "text", width: "40%", height: 14 }),
        /* @__PURE__ */ e(q, { variant: "text", width: "25%", height: 12 })
      ] })
    ] })
  ] });
}
function sa() {
  return /* @__PURE__ */ s("div", { className: "rounded-[var(--radius-lg)] border border-border bg-card overflow-hidden", children: [
    /* @__PURE__ */ e("div", { className: "px-4 py-3 border-b border-border bg-muted/30 flex gap-4", children: [120, 160, 100, 80].map((t, r) => /* @__PURE__ */ e(q, { variant: "text", width: t, height: 14 }, r)) }),
    [0, 1, 2, 3].map((t) => /* @__PURE__ */ s(
      "div",
      {
        className: "px-4 py-3 border-b border-border flex items-center gap-4",
        children: [
          /* @__PURE__ */ e(q, { variant: "circular", width: 28, height: 28 }),
          /* @__PURE__ */ e(q, { variant: "text", width: 100 + t * 15, height: 14 }),
          /* @__PURE__ */ e(
            q,
            {
              variant: "text",
              width: 140,
              height: 14,
              className: "flex-1"
            }
          ),
          /* @__PURE__ */ e(q, { variant: "rounded", width: 60, height: 22 }),
          /* @__PURE__ */ e(q, { variant: "text", width: 80, height: 14 })
        ]
      },
      t
    ))
  ] });
}
function la() {
  return /* @__PURE__ */ e("div", { className: "space-y-3 w-full max-w-md", children: [0, 1, 2].map((t) => /* @__PURE__ */ s(
    "div",
    {
      className: "flex items-center gap-3 px-3 py-2.5 rounded-[var(--radius-md)] border border-border bg-card",
      children: [
        /* @__PURE__ */ e(q, { variant: "circular", width: 40, height: 40 }),
        /* @__PURE__ */ s("div", { className: "flex-1 space-y-1.5", children: [
          /* @__PURE__ */ e(
            q,
            {
              variant: "text",
              width: `${60 + t * 10}%`,
              height: 14
            }
          ),
          /* @__PURE__ */ e(
            q,
            {
              variant: "text",
              width: `${40 + t * 5}%`,
              height: 12
            }
          )
        ] }),
        /* @__PURE__ */ e(q, { variant: "rounded", width: 50, height: 24 })
      ]
    },
    t
  )) });
}
const Zn = {
  sm: "w-4 h-4 border-2",
  md: "w-8 h-8 border-[3px]",
  lg: "w-12 h-12 border-4",
  xl: "w-16 h-16 border-4"
};
function Qn({ size: t = "md", color: r, className: o = "" }) {
  return /* @__PURE__ */ e(
    "div",
    {
      className: `${Zn[t]} rounded-full border-muted animate-spin ${o}`,
      style: { borderTopColor: r ?? "var(--primary)" },
      role: "status",
      "aria-label": "Loading"
    }
  );
}
Qn.displayName = "Spinner";
const Ve = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, Gt = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, qt = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, Jt = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
}, Pn = {
  sm: {
    fontFamily: "var(--font-h4)",
    fontSize: "var(--text-h4)",
    fontWeight: "var(--weight-h4)"
  },
  md: {
    fontFamily: "var(--font-h2)",
    fontSize: "var(--text-h2)",
    fontWeight: "var(--weight-h2)"
  },
  lg: {
    fontFamily: "var(--font-h1)",
    fontSize: "var(--text-h1)",
    fontWeight: "var(--weight-h1)"
  }
};
function ia({
  title: t,
  value: r,
  prefix: o,
  suffix: n,
  trend: a,
  trendLabel: i,
  icon: l,
  size: c = "md",
  loading: d
}) {
  const f = Pn[c], m = (a == null ? void 0 : a.direction) === "up" ? "text-chart-2" : (a == null ? void 0 : a.direction) === "down" ? "text-destructive" : "text-muted-foreground", v = (a == null ? void 0 : a.direction) === "up" ? Dt : (a == null ? void 0 : a.direction) === "down" ? At : Ee;
  return d ? /* @__PURE__ */ s("div", { className: "space-y-2 animate-pulse", children: [
    /* @__PURE__ */ e("div", { className: "h-3 w-20 bg-muted rounded-[var(--radius-sm)]" }),
    /* @__PURE__ */ e("div", { className: "h-8 w-32 bg-muted rounded-[var(--radius-sm)]" }),
    /* @__PURE__ */ e("div", { className: "h-3 w-24 bg-muted rounded-[var(--radius-sm)]" })
  ] }) : /* @__PURE__ */ s("div", { className: "space-y-1", children: [
    /* @__PURE__ */ s("div", { className: "flex items-center gap-2", children: [
      l && /* @__PURE__ */ e("span", { className: "text-muted-foreground", children: l }),
      /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: Ve, children: t })
    ] }),
    /* @__PURE__ */ s("div", { className: "flex items-baseline gap-1", children: [
      o && /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: Gt, children: o }),
      /* @__PURE__ */ e("span", { className: "text-foreground", style: f, children: typeof r == "number" ? r.toLocaleString() : r }),
      n && /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: Ve, children: n })
    ] }),
    a && /* @__PURE__ */ s("div", { className: `flex items-center gap-1 ${m}`, children: [
      /* @__PURE__ */ e(v, { size: 14 }),
      /* @__PURE__ */ s("span", { style: Jt, children: [
        a.value,
        "%"
      ] }),
      i && /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: qt, children: i })
    ] })
  ] });
}
function ca({
  title: t,
  value: r,
  prefix: o,
  trend: n,
  trendLabel: a,
  icon: i,
  iconBg: l
}) {
  const c = (n == null ? void 0 : n.direction) === "up" ? "text-chart-2" : (n == null ? void 0 : n.direction) === "down" ? "text-destructive" : "text-muted-foreground", d = (n == null ? void 0 : n.direction) === "up" ? Dt : (n == null ? void 0 : n.direction) === "down" ? At : Ee;
  return /* @__PURE__ */ e("div", { className: "rounded-[var(--radius-lg)] border border-border bg-card p-5", children: /* @__PURE__ */ s("div", { className: "flex items-start justify-between", children: [
    /* @__PURE__ */ s("div", { className: "space-y-1", children: [
      /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: Ve, children: t }),
      /* @__PURE__ */ s("div", { className: "flex items-baseline gap-1", children: [
        o && /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: Gt, children: o }),
        /* @__PURE__ */ e(
          "span",
          {
            className: "text-foreground",
            style: {
              fontFamily: "var(--font-h3)",
              fontSize: "var(--text-h3)",
              fontWeight: "var(--weight-h3)"
            },
            children: typeof r == "number" ? r.toLocaleString() : r
          }
        )
      ] }),
      n && /* @__PURE__ */ s("div", { className: `flex items-center gap-1 ${c}`, children: [
        /* @__PURE__ */ e(d, { size: 12 }),
        /* @__PURE__ */ s("span", { style: Jt, children: [
          n.value,
          "%"
        ] }),
        a && /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: qt, children: a })
      ] })
    ] }),
    i && /* @__PURE__ */ e(
      "div",
      {
        className: `w-10 h-10 rounded-[var(--radius-md)] flex items-center justify-center flex-shrink-0 ${l || "bg-primary/10 text-primary"}`,
        children: i
      }
    )
  ] }) });
}
const Vn = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, vt = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, xt = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, yt = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
};
function eo({
  steps: t,
  current: r,
  orientation: o = "horizontal",
  onStepClick: n,
  className: a = ""
}) {
  return o === "vertical" ? /* @__PURE__ */ e("div", { className: `flex flex-col ${a}`, children: t.map((i, l) => {
    const c = l < r, d = l === r;
    return /* @__PURE__ */ s(
      "div",
      {
        className: "flex gap-3",
        onClick: () => n == null ? void 0 : n(l),
        children: [
          /* @__PURE__ */ s("div", { className: "flex flex-col items-center", children: [
            /* @__PURE__ */ e(
              "div",
              {
                className: `w-8 h-8 rounded-full flex items-center justify-center flex-shrink-0 transition-colors ${n ? "cursor-pointer" : ""} ${c || d ? "bg-primary text-primary-foreground" : "bg-muted text-muted-foreground"}`,
                style: yt,
                children: c ? /* @__PURE__ */ e(fe, { size: 14 }) : l + 1
              }
            ),
            l < t.length - 1 && /* @__PURE__ */ e(
              "div",
              {
                className: `w-px flex-1 min-h-[24px] my-1 ${c ? "bg-primary" : "bg-border"}`
              }
            )
          ] }),
          /* @__PURE__ */ s("div", { className: `pb-6 ${l === t.length - 1 ? "pb-0" : ""}`, children: [
            /* @__PURE__ */ e(
              "span",
              {
                className: `block ${d ? "text-foreground" : "text-muted-foreground"}`,
                style: d ? vt : Vn,
                children: i.title
              }
            ),
            i.description && /* @__PURE__ */ e("span", { className: "text-muted-foreground block", style: xt, children: i.description })
          ] })
        ]
      },
      l
    );
  }) }) : /* @__PURE__ */ e("div", { className: `flex items-center ${a}`, children: t.map((i, l) => {
    const c = l < r, d = l === r;
    return /* @__PURE__ */ s(
      "div",
      {
        className: `flex items-center ${l < t.length - 1 ? "flex-1" : ""}`,
        children: [
          /* @__PURE__ */ s(
            "div",
            {
              className: "flex flex-col items-center gap-1",
              onClick: () => n == null ? void 0 : n(l),
              children: [
                /* @__PURE__ */ e(
                  "div",
                  {
                    className: `w-8 h-8 rounded-full flex items-center justify-center flex-shrink-0 transition-colors ${n ? "cursor-pointer" : ""} ${c || d ? "bg-primary text-primary-foreground" : "bg-muted text-muted-foreground"}`,
                    style: yt,
                    children: c ? /* @__PURE__ */ e(fe, { size: 14 }) : l + 1
                  }
                ),
                /* @__PURE__ */ e(
                  "span",
                  {
                    className: `text-center ${d ? "text-foreground" : "text-muted-foreground"}`,
                    style: d ? vt : xt,
                    children: i.title
                  }
                )
              ]
            }
          ),
          l < t.length - 1 && /* @__PURE__ */ e(
            "div",
            {
              className: `flex-1 h-px mx-2 mt-[-20px] ${c ? "bg-primary" : "bg-border"}`
            }
          )
        ]
      },
      l
    );
  }) });
}
eo.displayName = "Stepper";
const to = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, ro = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, no = {
  sm: { track: "w-7 h-4", thumb: "w-3 h-3", on: "translate-x-3" },
  md: { track: "w-10 h-6", thumb: "w-4.5 h-4.5", on: "translate-x-4.5" },
  lg: { track: "w-14 h-8", thumb: "w-6.5 h-6.5", on: "translate-x-6" }
}, oo = {
  primary: "bg-primary",
  success: "bg-chart-2",
  warning: "bg-chart-5",
  destructive: "bg-destructive"
}, ao = {
  sm: { w: 12, h: 12, on: 14 },
  md: { w: 18, h: 18, on: 18 },
  lg: { w: 26, h: 26, on: 24 }
};
function da({
  checked: t,
  onChange: r,
  label: o,
  description: n,
  size: a = "md",
  disabled: i,
  color: l = "primary"
}) {
  const c = no[a], d = ao[a];
  return /* @__PURE__ */ s(
    "label",
    {
      className: `inline-flex items-start gap-3 ${i ? "opacity-50 pointer-events-none" : "cursor-pointer"}`,
      children: [
        /* @__PURE__ */ e(
          "button",
          {
            role: "switch",
            "aria-checked": t,
            onClick: () => r(!t),
            disabled: i,
            className: `${c.track} rounded-full relative flex-shrink-0 transition-colors cursor-pointer ${t ? oo[l] : "bg-muted"}`,
            children: /* @__PURE__ */ e(
              "span",
              {
                className: "rounded-full bg-white shadow-sm absolute transition-transform",
                style: {
                  width: d.w,
                  height: d.h,
                  top: "50%",
                  transform: `translateY(-50%) translateX(${t ? d.on : 2}px)`,
                  left: 0
                }
              }
            )
          }
        ),
        (o || n) && /* @__PURE__ */ s("div", { children: [
          o && /* @__PURE__ */ e("span", { className: "text-foreground block", style: to, children: o }),
          n && /* @__PURE__ */ e("span", { className: "text-muted-foreground block", style: ro, children: n })
        ] })
      ]
    }
  );
}
const so = {
  default: "bg-muted/50 text-foreground border-border",
  primary: "bg-primary/10 text-primary border-primary/20",
  success: "bg-chart-2/10 text-chart-2 border-chart-2/20",
  warning: "bg-chart-5/10 text-chart-5 border-chart-5/20",
  destructive: "bg-destructive/10 text-destructive border-destructive/20",
  info: "bg-accent text-accent-foreground border-primary/20"
}, lo = {
  sm: "px-1.5 py-0.5",
  md: "px-2 py-0.5",
  lg: "px-3 py-1.5"
}, io = {
  sm: {
    fontFamily: "var(--font-button)",
    fontSize: "var(--text-button)",
    fontWeight: "var(--weight-button)"
  },
  md: {
    fontFamily: "var(--font-button)",
    fontSize: "var(--text-button)",
    fontWeight: "var(--weight-button)"
  },
  lg: {
    fontFamily: "var(--font-label)",
    fontSize: "var(--text-label)",
    fontWeight: "var(--weight-button)"
  }
};
function co({
  children: t,
  color: r = "default",
  size: o = "md",
  icon: n,
  closable: a,
  onClose: i,
  className: l = ""
}) {
  const c = so[r], d = lo[o], f = io[o];
  return /* @__PURE__ */ s(
    "span",
    {
      className: `inline-flex items-center gap-1 rounded-full border ${c} ${d} ${l}`,
      style: f,
      children: [
        n && /* @__PURE__ */ e("span", { className: "flex-shrink-0", children: n }),
        t,
        a && /* @__PURE__ */ e("button", { className: "hover:opacity-70 cursor-pointer", onClick: i, children: /* @__PURE__ */ e(J, { size: o === "sm" ? 10 : 12 }) })
      ]
    }
  );
}
co.displayName = "Tag";
const uo = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, fo = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
}, mo = {
  default: "bg-card border-border",
  outline: "bg-transparent border-border",
  filled: "bg-muted/30 border-transparent"
};
function ho({
  tags: t,
  onChange: r,
  placeholder: o,
  disabled: n,
  maxTags: a,
  variant: i = "default",
  className: l = ""
}) {
  const [c, d] = z(""), f = () => {
    const v = c.trim();
    !v || t.includes(v) || a && t.length >= a || (r([...t, v]), d(""));
  }, m = (v) => {
    r(t.filter((p) => p !== v));
  };
  return /* @__PURE__ */ s(
    "div",
    {
      className: `flex flex-wrap items-center gap-1.5 px-3 py-2 rounded-[var(--radius)] border ${mo[i]} transition-colors focus-within:border-primary ${n ? "opacity-50 pointer-events-none" : ""} ${l}`,
      children: [
        t.map((v) => /* @__PURE__ */ s(
          "span",
          {
            className: "inline-flex items-center gap-1 px-2 py-0.5 rounded-[var(--radius-sm)] bg-primary/10 text-primary",
            style: fo,
            children: [
              v,
              /* @__PURE__ */ e(
                "button",
                {
                  className: "hover:text-destructive cursor-pointer",
                  onClick: () => m(v),
                  children: /* @__PURE__ */ e(J, { size: 12 })
                }
              )
            ]
          },
          v
        )),
        /* @__PURE__ */ e(
          "input",
          {
            value: c,
            onChange: (v) => d(v.target.value),
            onKeyDown: (v) => {
              v.key === "Enter" && (v.preventDefault(), f()), v.key === "Backspace" && !c && t.length && r(t.slice(0, -1));
            },
            placeholder: t.length ? "" : o,
            className: "flex-1 min-w-[80px] bg-transparent outline-none text-foreground placeholder:text-muted-foreground",
            style: uo,
            disabled: n
          }
        )
      ]
    }
  );
}
ho.displayName = "TagInput";
const bo = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, wt = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, Je = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
};
function ua({
  items: t,
  variant: r = "default",
  size: o = "md"
}) {
  const n = o === "sm" ? "w-6 h-6" : o === "lg" ? "w-10 h-10" : "w-8 h-8", a = o === "sm" ? 12 : o === "lg" ? 18 : 14, i = (l) => l === "completed" ? "bg-chart-2 text-white" : l === "current" ? "bg-primary text-primary-foreground" : l === "error" ? "bg-destructive text-white" : "bg-muted text-muted-foreground";
  return r === "compact" ? /* @__PURE__ */ e("div", { className: "space-y-0", children: t.map((l, c) => /* @__PURE__ */ s("div", { className: "flex items-start gap-3", children: [
    /* @__PURE__ */ s("div", { className: "flex flex-col items-center", children: [
      /* @__PURE__ */ e(
        "div",
        {
          className: `${n} rounded-full flex items-center justify-center flex-shrink-0 ${i(
            l.status
          )}`,
          children: l.icon || (l.status === "completed" ? /* @__PURE__ */ e(se, { size: a }) : /* @__PURE__ */ e(ct, { size: a }))
        }
      ),
      c < t.length - 1 && /* @__PURE__ */ e("div", { className: "w-0.5 flex-1 min-h-4 bg-border" })
    ] }),
    /* @__PURE__ */ s("div", { className: "pb-4 min-w-0", children: [
      /* @__PURE__ */ s("div", { className: "flex items-center gap-2", children: [
        /* @__PURE__ */ e("span", { className: "text-foreground", style: wt, children: l.title }),
        l.time && /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: Je, children: l.time })
      ] }),
      l.description && /* @__PURE__ */ e(
        "p",
        {
          className: "text-muted-foreground mt-0.5",
          style: Je,
          children: l.description
        }
      )
    ] })
  ] }, c)) }) : /* @__PURE__ */ e("div", { className: "relative", children: t.map((l, c) => {
    const d = r === "alternate" && c % 2 === 1;
    return /* @__PURE__ */ s(
      "div",
      {
        className: `flex gap-4 pb-8 last:pb-0 ${d ? "flex-row-reverse" : ""}`,
        children: [
          /* @__PURE__ */ s("div", { className: "flex flex-col items-center flex-shrink-0", children: [
            /* @__PURE__ */ e(
              "div",
              {
                className: `${n} rounded-full flex items-center justify-center ${i(
                  l.status
                )}`,
                children: l.icon || (l.status === "completed" ? /* @__PURE__ */ e(se, { size: a }) : /* @__PURE__ */ e(ct, { size: a }))
              }
            ),
            c < t.length - 1 && /* @__PURE__ */ e("div", { className: "w-0.5 flex-1 bg-border mt-1" })
          ] }),
          /* @__PURE__ */ s(
            "div",
            {
              className: `flex-1 min-w-0 ${d ? "text-right" : ""}`,
              children: [
                /* @__PURE__ */ s(
                  "div",
                  {
                    className: `flex items-center gap-2 ${d ? "justify-end" : ""}`,
                    children: [
                      /* @__PURE__ */ e("span", { className: "text-foreground", style: wt, children: l.title }),
                      l.time && /* @__PURE__ */ e("span", { className: "text-muted-foreground", style: Je, children: l.time })
                    ]
                  }
                ),
                l.description && /* @__PURE__ */ e(
                  "p",
                  {
                    className: "text-muted-foreground mt-0.5",
                    style: bo,
                    children: l.description
                  }
                ),
                l.children && /* @__PURE__ */ e("div", { className: "mt-2", children: l.children })
              ]
            }
          )
        ]
      },
      c
    );
  }) });
}
const go = {
  top: "bottom-full left-1/2 -translate-x-1/2 mb-2",
  bottom: "top-full left-1/2 -translate-x-1/2 mt-2",
  left: "right-full top-1/2 -translate-y-1/2 mr-2",
  right: "left-full top-1/2 -translate-y-1/2 ml-2"
}, po = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
};
function vo({
  children: t,
  content: r,
  placement: o = "top",
  className: n = ""
}) {
  const [a, i] = z(!1);
  return /* @__PURE__ */ s(
    "div",
    {
      className: `relative inline-flex ${n}`,
      onMouseEnter: () => i(!0),
      onMouseLeave: () => i(!1),
      onFocus: () => i(!0),
      onBlur: () => i(!1),
      children: [
        t,
        a && /* @__PURE__ */ e(
          "div",
          {
            className: `absolute ${go[o]} z-50 px-3 py-1.5 rounded-[var(--radius-sm)] bg-foreground text-background whitespace-nowrap shadow-md`,
            style: po,
            role: "tooltip",
            children: r
          }
        )
      ]
    }
  );
}
vo.displayName = "Tooltip";
const Ae = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-button)"
};
function xo({
  brand: t,
  breadcrumbs: r,
  actions: o,
  user: n,
  height: a = "72px",
  showSearch: i = !1,
  searchPlaceholder: l = "Search... (Ctrl+K)",
  onSearchClick: c,
  notificationCount: d,
  onNotificationClick: f,
  onMobileMenuClick: m,
  onUserClick: v,
  onBreadcrumbClick: p,
  className: u = ""
}) {
  const S = n != null && n.name ? n.name.split(" ").map((N) => N[0]).join("").slice(0, 2).toUpperCase() : "";
  return /* @__PURE__ */ s(
    "div",
    {
      className: `bg-card border-b border-border flex items-center px-5 gap-4 ${u}`,
      style: { height: a },
      children: [
        m && /* @__PURE__ */ e(
          "button",
          {
            className: "lg:hidden w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] text-foreground hover:bg-muted/30 transition-colors cursor-pointer",
            onClick: m,
            children: /* @__PURE__ */ e(Nr, { size: 16 })
          }
        ),
        t && /* @__PURE__ */ e(Qe, { children: typeof t.logo == "string" ? /* @__PURE__ */ e(
          "img",
          {
            src: t.logo,
            alt: t.name,
            className: "w-8 h-8 rounded-[var(--radius-sm)] flex-shrink-0 object-cover"
          }
        ) : t.logo ? /* @__PURE__ */ e("div", { className: "w-8 h-8 rounded-[var(--radius-sm)] overflow-hidden flex-shrink-0", children: t.logo }) : /* @__PURE__ */ e(
          "div",
          {
            className: "w-8 h-8 rounded-[var(--radius-sm)] bg-primary flex items-center justify-center text-primary-foreground flex-shrink-0",
            style: Ae,
            children: t.name.charAt(0).toUpperCase()
          }
        ) }),
        r && r.length > 0 && /* @__PURE__ */ e("nav", { className: "hidden sm:flex items-center gap-1.5 text-muted-foreground", style: Ae, children: r.map((N, h) => /* @__PURE__ */ s(et.Fragment, { children: [
          h > 0 && /* @__PURE__ */ e(le, { size: 12 }),
          N.href || p ? /* @__PURE__ */ e(
            "span",
            {
              className: `cursor-pointer ${h === r.length - 1 ? "text-foreground" : "hover:text-foreground"}`,
              onClick: () => p == null ? void 0 : p(N, h),
              children: N.label
            }
          ) : /* @__PURE__ */ e(
            "span",
            {
              className: h === r.length - 1 ? "text-foreground" : "",
              children: N.label
            }
          )
        ] }, h)) }),
        /* @__PURE__ */ e("div", { className: "flex-1" }),
        i && /* @__PURE__ */ s(
          "div",
          {
            className: "hidden md:flex items-center gap-2 px-3 h-8 rounded-[var(--radius)] border border-border bg-muted/20 text-muted-foreground w-56 cursor-pointer hover:border-primary/40 transition-colors",
            onClick: c,
            children: [
              /* @__PURE__ */ e(Be, { size: 14 }),
              /* @__PURE__ */ e("span", { style: Ae, children: l })
            ]
          }
        ),
        o,
        f && /* @__PURE__ */ s(
          "button",
          {
            className: "relative w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] text-muted-foreground hover:text-foreground hover:bg-muted/30 transition-colors cursor-pointer",
            onClick: f,
            children: [
              /* @__PURE__ */ e(Mt, { size: 16 }),
              d !== void 0 && d > 0 && /* @__PURE__ */ e("span", { className: "absolute top-1 right-1 w-2 h-2 rounded-full bg-destructive" })
            ]
          }
        ),
        n && /* @__PURE__ */ e(
          "div",
          {
            className: "cursor-pointer",
            onClick: v,
            children: n.avatar ? /* @__PURE__ */ e(
              "img",
              {
                src: n.avatar,
                alt: n.name,
                className: "w-8 h-8 rounded-full object-cover"
              }
            ) : /* @__PURE__ */ e(
              "div",
              {
                className: "w-8 h-8 rounded-full bg-primary text-primary-foreground flex items-center justify-center",
                style: Ae,
                children: S
              }
            )
          }
        )
      ]
    }
  );
}
xo.displayName = "TopNavbar";
const yo = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, wo = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, Ze = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
};
function Nt({
  title: t,
  items: r,
  selected: o,
  onToggle: n,
  search: a,
  onSearch: i,
  searchable: l
}) {
  return /* @__PURE__ */ s("div", { className: "flex-1 min-w-[180px] rounded-[var(--radius-lg)] border border-border bg-card overflow-hidden", children: [
    /* @__PURE__ */ s("div", { className: "px-3 py-2.5 border-b border-border bg-muted/30 flex items-center justify-between", children: [
      /* @__PURE__ */ e("span", { className: "text-foreground", style: wo, children: t }),
      /* @__PURE__ */ s("span", { className: "text-muted-foreground", style: Ze, children: [
        r.length,
        " items"
      ] })
    ] }),
    l && /* @__PURE__ */ s("div", { className: "px-3 py-2 border-b border-border flex items-center gap-2", children: [
      /* @__PURE__ */ e(Be, { size: 12, className: "text-muted-foreground" }),
      /* @__PURE__ */ e(
        "input",
        {
          value: a,
          onChange: (c) => i(c.target.value),
          placeholder: "Search\\u2026",
          className: "flex-1 bg-transparent outline-none text-foreground placeholder:text-muted-foreground",
          style: Ze
        }
      )
    ] }),
    /* @__PURE__ */ s("div", { className: "max-h-[240px] overflow-y-auto py-1", children: [
      r.length === 0 && /* @__PURE__ */ e(
        "div",
        {
          className: "px-3 py-6 text-center text-muted-foreground",
          style: Ze,
          children: "No items"
        }
      ),
      r.map((c) => /* @__PURE__ */ s(
        "button",
        {
          onClick: () => !c.disabled && n(c.id),
          className: `w-full flex items-center gap-2 px-3 py-1.5 transition-colors cursor-pointer
              ${o.has(c.id) ? "bg-primary/10" : "hover:bg-accent"}
              ${c.disabled ? "opacity-40 pointer-events-none" : ""}`,
          style: yo,
          children: [
            /* @__PURE__ */ e(
              "span",
              {
                className: `w-4 h-4 rounded-[var(--radius-sm)] border flex items-center justify-center flex-shrink-0
              ${o.has(c.id) ? "bg-primary border-primary" : "border-border"}`,
                children: o.has(c.id) && /* @__PURE__ */ e("span", { className: "w-2 h-2 rounded-[1px] bg-primary-foreground" })
              }
            ),
            /* @__PURE__ */ e("span", { className: "text-foreground truncate", children: c.label })
          ]
        },
        c.id
      ))
    ] })
  ] });
}
function fa({
  sourceTitle: t = "Available",
  targetTitle: r = "Selected",
  items: o,
  defaultTarget: n = [],
  searchable: a,
  value: i,
  onChange: l
}) {
  const [c, d] = z(
    new Set(n)
  ), f = i !== void 0 ? new Set(i) : c, m = (F) => {
    l && l(Array.from(F)), i === void 0 && d(F);
  }, [v, p] = z(
    /* @__PURE__ */ new Set()
  ), [u, S] = z(
    /* @__PURE__ */ new Set()
  ), [N, h] = z(""), [x, D] = z(""), b = o.filter((F) => !f.has(F.id)), y = o.filter((F) => f.has(F.id)), $ = N ? b.filter(
    (F) => F.label.toLowerCase().includes(N.toLowerCase())
  ) : b, k = x ? y.filter(
    (F) => F.label.toLowerCase().includes(x.toLowerCase())
  ) : y, M = () => {
    const F = new Set(f);
    v.forEach((w) => F.add(w)), m(F), p(/* @__PURE__ */ new Set());
  }, I = () => {
    const F = new Set(f);
    u.forEach((w) => F.delete(w)), m(F), S(/* @__PURE__ */ new Set());
  }, W = () => {
    m(new Set(o.filter((F) => !F.disabled).map((F) => F.id))), p(/* @__PURE__ */ new Set());
  }, L = () => {
    m(/* @__PURE__ */ new Set()), S(/* @__PURE__ */ new Set());
  }, Y = (F, w, T) => {
    const H = new Set(w);
    H.has(F) ? H.delete(F) : H.add(F), T(H);
  };
  return /* @__PURE__ */ s("div", { className: "flex items-stretch gap-3 flex-wrap", children: [
    /* @__PURE__ */ e(
      Nt,
      {
        title: t,
        items: $,
        selected: v,
        onToggle: (F) => Y(F, v, p),
        search: N,
        onSearch: h,
        searchable: a
      }
    ),
    /* @__PURE__ */ s("div", { className: "flex flex-col items-center justify-center gap-1.5 py-4", children: [
      /* @__PURE__ */ e(
        "button",
        {
          onClick: W,
          disabled: b.length === 0,
          className: "w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] border border-border text-muted-foreground hover:text-foreground hover:bg-accent disabled:opacity-30 transition-colors cursor-pointer",
          children: /* @__PURE__ */ e(rt, { size: 14 })
        }
      ),
      /* @__PURE__ */ e(
        "button",
        {
          onClick: M,
          disabled: v.size === 0,
          className: "w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] border border-border text-muted-foreground hover:text-foreground hover:bg-accent disabled:opacity-30 transition-colors cursor-pointer",
          children: /* @__PURE__ */ e(le, { size: 14 })
        }
      ),
      /* @__PURE__ */ e(
        "button",
        {
          onClick: I,
          disabled: u.size === 0,
          className: "w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] border border-border text-muted-foreground hover:text-foreground hover:bg-accent disabled:opacity-30 transition-colors cursor-pointer",
          children: /* @__PURE__ */ e($e, { size: 14 })
        }
      ),
      /* @__PURE__ */ e(
        "button",
        {
          onClick: L,
          disabled: y.length === 0,
          className: "w-8 h-8 flex items-center justify-center rounded-[var(--radius-md)] border border-border text-muted-foreground hover:text-foreground hover:bg-accent disabled:opacity-30 transition-colors cursor-pointer",
          children: /* @__PURE__ */ e(tt, { size: 14 })
        }
      )
    ] }),
    /* @__PURE__ */ e(
      Nt,
      {
        title: r,
        items: k,
        selected: u,
        onToggle: (F) => Y(F, u, S),
        search: x,
        onSearch: D,
        searchable: a
      }
    )
  ] });
}
const No = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
};
function Zt({
  node: t,
  level: r,
  expanded: o,
  selected: n,
  onToggle: a,
  onSelect: i,
  selectable: l,
  showLines: c
}) {
  const d = t.children && t.children.length > 0, f = o.has(t.id), m = n.has(t.id), v = d ? f ? /* @__PURE__ */ e(Sr, { size: 14, className: "text-chart-5" }) : /* @__PURE__ */ e($r, { size: 14, className: "text-chart-5" }) : /* @__PURE__ */ e(Ct, { size: 14, className: "text-muted-foreground" });
  return /* @__PURE__ */ s("div", { children: [
    /* @__PURE__ */ s(
      "div",
      {
        className: `flex items-center gap-1.5 py-1 px-2 rounded-[var(--radius-sm)] transition-colors
          ${t.disabled ? "opacity-40 pointer-events-none" : "cursor-pointer hover:bg-accent"}
          ${m ? "bg-primary/10 text-primary" : "text-foreground"}`,
        style: { paddingLeft: `${r * 20 + 8}px`, ...No },
        onClick: () => {
          d && a(t.id), l && i(t.id);
        },
        children: [
          d ? /* @__PURE__ */ e("span", { className: "w-4 h-4 flex items-center justify-center flex-shrink-0", children: f ? /* @__PURE__ */ e(Ye, { size: 12 }) : /* @__PURE__ */ e(le, { size: 12 }) }) : /* @__PURE__ */ e("span", { className: "w-4" }),
          l && /* @__PURE__ */ e(
            "span",
            {
              className: `w-4 h-4 rounded-[var(--radius-sm)] border flex items-center justify-center flex-shrink-0 ${m ? "bg-primary border-primary" : "border-border"}`,
              children: m && /* @__PURE__ */ e("span", { className: "w-2 h-2 rounded-[1px] bg-primary-foreground" })
            }
          ),
          /* @__PURE__ */ e("span", { className: "flex-shrink-0", children: t.icon || v }),
          /* @__PURE__ */ e("span", { className: "truncate", children: t.label })
        ]
      }
    ),
    d && f && /* @__PURE__ */ e("div", { className: c ? "border-l border-border ml-[22px]" : "", children: t.children.map((p) => /* @__PURE__ */ e(
      Zt,
      {
        node: p,
        level: r + 1,
        expanded: o,
        selected: n,
        onToggle: a,
        onSelect: i,
        selectable: l,
        showLines: c
      },
      p.id
    )) })
  ] });
}
function ma({
  data: t,
  selectable: r,
  showLines: o,
  defaultExpanded: n,
  expandedItems: a,
  onExpandChange: i,
  selectedItems: l,
  onSelect: c
}) {
  const [d, f] = z(
    new Set(n || [])
  ), [m, v] = z(/* @__PURE__ */ new Set()), p = a !== void 0 ? new Set(a) : d, u = l !== void 0 ? new Set(l) : m, S = (h) => {
    const x = new Set(p);
    x.has(h) ? x.delete(h) : x.add(h), i && i(Array.from(x)), a === void 0 && f(x);
  }, N = (h) => {
    const x = new Set(u);
    x.has(h) ? x.delete(h) : x.add(h), c && c(Array.from(x)), l === void 0 && v(x);
  };
  return /* @__PURE__ */ e("div", { className: "rounded-[var(--radius-md)] border border-border bg-card py-1", children: t.map((h) => /* @__PURE__ */ e(
    Zt,
    {
      node: h,
      level: 0,
      expanded: p,
      selected: u,
      onToggle: S,
      onSelect: N,
      selectable: r,
      showLines: o
    },
    h.id
  )) });
}
const So = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, Qt = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, Pt = zt({ name: "" });
function ha() {
  return $t(Pt);
}
function $o({
  htmlFor: t,
  required: r,
  children: o,
  className: n = ""
}) {
  return /* @__PURE__ */ s(
    "label",
    {
      htmlFor: t,
      className: `block text-foreground mb-1.5 ${n}`,
      style: So,
      children: [
        o,
        r && /* @__PURE__ */ e("span", { className: "text-destructive ml-0.5", children: "*" })
      ]
    }
  );
}
function zo({ message: t, className: r = "" }) {
  return t ? /* @__PURE__ */ s(
    "div",
    {
      className: `flex items-center gap-1 mt-1.5 text-destructive ${r}`,
      style: Qt,
      role: "alert",
      children: [
        /* @__PURE__ */ e(ke, { size: 12, className: "flex-shrink-0" }),
        /* @__PURE__ */ e("span", { children: t })
      ]
    }
  ) : null;
}
function ko({ children: t, className: r = "" }) {
  return /* @__PURE__ */ e(
    "p",
    {
      className: `mt-1.5 text-muted-foreground ${r}`,
      style: Qt,
      children: t
    }
  );
}
function ba({
  name: t,
  label: r,
  required: o,
  error: n,
  helperText: a,
  children: i,
  className: l = ""
}) {
  return /* @__PURE__ */ e(Pt.Provider, { value: { name: t, error: n }, children: /* @__PURE__ */ s("div", { className: `${l}`, children: [
    r && /* @__PURE__ */ e($o, { htmlFor: t, required: o, children: r }),
    i,
    n ? /* @__PURE__ */ e(zo, { message: n }) : a ? /* @__PURE__ */ e(ko, { children: a }) : null
  ] }) });
}
const Fo = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, Wo = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-label)"
}, Lo = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, jo = {
  sm: "h-8 text-sm",
  md: "h-10",
  lg: "h-12 text-lg"
}, St = {
  sm: 14,
  md: 16,
  lg: 18
};
function ga({
  value: t,
  defaultValue: r = 0,
  onChange: o,
  min: n,
  max: a,
  step: i = 1,
  disabled: l,
  size: c = "md",
  label: d,
  placeholder: f,
  error: m,
  className: v = ""
}) {
  const [p, u] = z(r), S = t ?? p, N = (D) => {
    const b = Math.max(n ?? -1 / 0, Math.min(a ?? 1 / 0, D));
    o && o(b), t === void 0 && u(b);
  }, h = n === void 0 || S - i >= n, x = a === void 0 || S + i <= a;
  return /* @__PURE__ */ s("div", { className: v, children: [
    d && /* @__PURE__ */ e("label", { className: "block text-foreground mb-1.5", style: Fo, children: d }),
    /* @__PURE__ */ s(
      "div",
      {
        className: `inline-flex items-center rounded-[var(--radius-md)] border ${m ? "border-destructive" : "border-border"} bg-background overflow-hidden ${l ? "opacity-50 pointer-events-none" : ""}`,
        children: [
          /* @__PURE__ */ e(
            "button",
            {
              type: "button",
              onClick: () => N(S - i),
              disabled: l || !h,
              className: "flex items-center justify-center px-2.5 border-r border-border text-muted-foreground hover:text-foreground hover:bg-accent disabled:opacity-30 transition-colors cursor-pointer",
              style: { height: "100%" },
              children: /* @__PURE__ */ e(Ee, { size: St[c] })
            }
          ),
          /* @__PURE__ */ e(
            "input",
            {
              type: "number",
              value: S,
              placeholder: f,
              onChange: (D) => {
                const b = parseFloat(D.target.value);
                isNaN(b) || N(b);
              },
              disabled: l,
              className: `w-20 text-center bg-transparent outline-none text-foreground ${jo[c]} [&::-webkit-inner-spin-button]:appearance-none [&::-webkit-outer-spin-button]:appearance-none`,
              style: Wo
            }
          ),
          /* @__PURE__ */ e(
            "button",
            {
              type: "button",
              onClick: () => N(S + i),
              disabled: l || !x,
              className: "flex items-center justify-center px-2.5 border-l border-border text-muted-foreground hover:text-foreground hover:bg-accent disabled:opacity-30 transition-colors cursor-pointer",
              style: { height: "100%" },
              children: /* @__PURE__ */ e(jt, { size: St[c] })
            }
          )
        ]
      }
    ),
    m && /* @__PURE__ */ e("p", { className: "mt-1.5 text-destructive", style: Lo, children: m })
  ] });
}
const Co = {
  fontFamily: "var(--font-label)",
  fontSize: "var(--text-label)",
  fontWeight: "var(--weight-button)"
}, Mo = {
  fontFamily: "var(--font-button)",
  fontSize: "var(--text-button)",
  fontWeight: "var(--weight-label)"
}, Do = {
  sm: "w-9 h-9 text-sm",
  md: "w-11 h-11 text-base",
  lg: "w-14 h-14 text-xl"
};
function pa({
  length: t = 6,
  value: r,
  onChange: o,
  onComplete: n,
  disabled: a,
  error: i,
  size: l = "md",
  label: c,
  masked: d,
  className: f = ""
}) {
  const [m, v] = z(() => "".padEnd(t, "")), p = X([]), S = (r ?? m).padEnd(t, "").split("").slice(0, t), N = P(
    (b) => {
      o && o(b), r === void 0 && v(b), b.replace(/ /g, "").length === t && n && n(b);
    },
    [o, n, r, t]
  ), h = (b, y) => {
    var M;
    if (!/^\d$/.test(y)) return;
    const $ = S.slice();
    $[b] = y;
    const k = $.join("");
    N(k), b < t - 1 && ((M = p.current[b + 1]) == null || M.focus());
  }, x = (b, y) => {
    var $, k, M;
    if (y.key === "Backspace") {
      y.preventDefault();
      const I = S.slice();
      I[b] !== " " && I[b] !== "" ? (I[b] = " ", N(I.join(""))) : b > 0 && (I[b - 1] = " ", N(I.join("")), ($ = p.current[b - 1]) == null || $.focus());
    } else y.key === "ArrowLeft" && b > 0 ? (k = p.current[b - 1]) == null || k.focus() : y.key === "ArrowRight" && b < t - 1 && ((M = p.current[b + 1]) == null || M.focus());
  };
  return /* @__PURE__ */ s("div", { className: f, children: [
    c && /* @__PURE__ */ e("label", { className: "block text-foreground mb-2", style: Co, children: c }),
    /* @__PURE__ */ e("div", { className: "flex gap-2", onPaste: (b) => {
      var $;
      b.preventDefault();
      const y = b.clipboardData.getData("text").replace(/\D/g, "").slice(0, t);
      if (y) {
        N(y.padEnd(t, " "));
        const k = Math.min(y.length, t - 1);
        ($ = p.current[k]) == null || $.focus();
      }
    }, children: S.map((b, y) => /* @__PURE__ */ e(
      "input",
      {
        ref: ($) => {
          p.current[y] = $;
        },
        type: d ? "password" : "text",
        inputMode: "numeric",
        maxLength: 1,
        value: b === " " ? "" : b,
        disabled: a,
        onChange: ($) => {
          const k = $.target.value;
          k && h(y, k[k.length - 1]);
        },
        onKeyDown: ($) => x(y, $),
        onFocus: ($) => $.target.select(),
        className: `${Do[l]} text-center rounded-[var(--radius-md)] border outline-none transition-all
              ${i ? "border-destructive focus:ring-2 focus:ring-destructive/30" : "border-border focus:border-primary focus:ring-2 focus:ring-primary/20"}
              ${a ? "opacity-50 pointer-events-none bg-muted" : "bg-background"}
              text-foreground`,
        style: { fontFamily: "var(--font-button)", fontWeight: "var(--weight-button)" }
      },
      y
    )) }),
    i && /* @__PURE__ */ e("p", { className: "mt-2 text-destructive", style: Mo, children: i })
  ] });
}
export {
  Vr as Accordion,
  Ht as AccordionItem,
  Mr as Alert,
  on as Avatar,
  an as AvatarGroup,
  _o as Badge,
  Qo as Breadcrumb,
  Fr as Button,
  Bo as ButtonGroup,
  dn as Card,
  fn as CardBody,
  mn as CardFooter,
  un as CardHeader,
  Ro as Checkbox,
  Ho as CheckboxGroup,
  Po as ColorPicker,
  Xo as ConfirmDialog,
  Fr as DSButton,
  Ro as DSCheckbox,
  Lr as DSInput,
  Oo as DSRadio,
  Jo as DSTable,
  jr as DSTextarea,
  Ko as DatePicker,
  yn as Divider,
  Nn as Drawer,
  kn as Dropdown,
  Vo as EmptyState,
  ea as FileUpload,
  zo as FormError,
  ba as FormField,
  ko as FormHelperText,
  $o as FormLabel,
  Wr as IconButton,
  ta as ImagePreview,
  Lr as Input,
  En as Menu,
  Ur as Modal,
  Yn as Notification,
  ra as NotificationCenter,
  ga as NumberInput,
  pa as OTPInput,
  Go as Pagination,
  na as Popover,
  On as ProgressBar,
  Oo as Radio,
  Uo as RadioGroup,
  oa as Rating,
  qo as SearchField,
  Jn as Sidebar,
  q as Skeleton,
  aa as SkeletonCard,
  la as SkeletonList,
  sa as SkeletonTable,
  Qn as Spinner,
  ca as StatCard,
  ia as Statistic,
  eo as Stepper,
  da as Switch,
  Jo as Table,
  Zo as Tabs,
  co as Tag,
  ho as TagInput,
  jr as Textarea,
  ua as Timeline,
  Yo as ToastContainer,
  vo as Tooltip,
  xo as TopNavbar,
  fa as TransferList,
  ma as Tree,
  ve as toast,
  ha as useFormField
};
